From e4a807697fd4aea229dda7587bc2422d8ad004fa Mon Sep 17 00:00:00 2001
From: Mingcong Bai <baimingcong@uniontech.com>
Date: Wed, 24 Jul 2024 14:38:45 +0800
Subject: [PATCH 179/360] arch: arm64: loongarch: x86: clean up release-bound
 kernel configs

Drop debug, unit tests, self-tests, error/fault injection, emulated/fake
drivers, and non-critical tracing support from the release kernel configs
to save space and unneeded debug outputs (that may confuse our users).

With the exception of DebugFS support for rtw88/89, as we maintain a set
of vendor patches for these devices and debug outputs are not enabled
by default unless rtw8{8,9}_core.debug_mask=0x7fff was specificed.

Signed-off-by: Mingcong Bai <baimingcong@uniontech.com>
---
 .../configs/deepin_arm64_desktop_defconfig    | 461 +++++++++++-------
 .../deepin_loongarch_desktop_defconfig        |  81 ++-
 arch/x86/configs/deepin_x86_desktop_defconfig | 130 +----
 3 files changed, 346 insertions(+), 326 deletions(-)

diff --git a/arch/arm64/configs/deepin_arm64_desktop_defconfig b/arch/arm64/configs/deepin_arm64_desktop_defconfig
index 3b5926ffaafe..427be175a870 100644
--- a/arch/arm64/configs/deepin_arm64_desktop_defconfig
+++ b/arch/arm64/configs/deepin_arm64_desktop_defconfig
@@ -1,4 +1,4 @@
-CONFIG_COMPILE_TEST=y
+CONFIG_WERROR=y
 CONFIG_BUILD_SALT="-arm64-desktop-rolling"
 CONFIG_KERNEL_ZSTD=y
 CONFIG_DEFAULT_HOSTNAME=""
@@ -110,13 +110,11 @@ CONFIG_CMDLINE="pcie_aspm=off"
 CONFIG_SUSPEND_SKIP_SYNC=y
 CONFIG_HIBERNATION=y
 CONFIG_PM_STD_PARTITION="/dev/sda2"
-CONFIG_PM_DEBUG=y
 CONFIG_ENERGY_MODEL=y
 CONFIG_CPU_IDLE_GOV_LADDER=y
 CONFIG_CPU_IDLE_GOV_MENU=y
 CONFIG_CPU_IDLE_GOV_TEO=y
 CONFIG_ARM_PSCI_CPUIDLE=y
-CONFIG_ARM_CLPS711X_CPUIDLE=y
 CONFIG_CPU_FREQ=y
 CONFIG_CPU_FREQ_STAT=y
 CONFIG_CPU_FREQ_DEFAULT_GOV_USERSPACE=y
@@ -133,12 +131,10 @@ CONFIG_ARM_SCPI_CPUFREQ=m
 CONFIG_ARM_IMX_CPUFREQ_DT=m
 CONFIG_ARM_MEDIATEK_CPUFREQ=m
 CONFIG_ARM_QCOM_CPUFREQ_HW=m
-CONFIG_ARM_RASPBERRYPI_CPUFREQ=m
 CONFIG_ARM_SCMI_CPUFREQ=m
 CONFIG_QORIQ_CPUFREQ=m
 CONFIG_ACPI=y
 CONFIG_ACPI_FPDT=y
-CONFIG_ACPI_EC_DEBUGFS=m
 CONFIG_ACPI_BUTTON=m
 CONFIG_ACPI_TINY_POWER_BUTTON=m
 CONFIG_ACPI_FAN=m
@@ -224,6 +220,7 @@ CONFIG_DEFAULT_MMAP_MIN_ADDR=65536
 CONFIG_MEMORY_FAILURE=y
 CONFIG_TRANSPARENT_HUGEPAGE=y
 CONFIG_TRANSPARENT_HUGEPAGE_MADVISE=y
+CONFIG_CMA=y
 CONFIG_ZONE_DEVICE=y
 CONFIG_USERFAULTFD=y
 CONFIG_LRU_GEN=y
@@ -664,6 +661,7 @@ CONFIG_BT_6LOWPAN=m
 CONFIG_BT_LEDS=y
 CONFIG_BT_MSFTEXT=y
 CONFIG_BT_AOSPEXT=y
+# CONFIG_BT_DEBUGFS is not set
 CONFIG_BT_HCIBTUSB=m
 CONFIG_BT_HCIBTUSB_AUTOSUSPEND=y
 CONFIG_BT_HCIBTUSB_MTK=y
@@ -688,7 +686,6 @@ CONFIG_BT_MRVL_SDIO=m
 CONFIG_BT_ATH3K=m
 CONFIG_BT_MTKSDIO=m
 CONFIG_BT_MTKUART=m
-CONFIG_BT_QCOMSMD=m
 CONFIG_BT_VIRTIO=m
 CONFIG_BT_NXPUART=m
 CONFIG_AF_RXRPC_IPV6=y
@@ -752,22 +749,21 @@ CONFIG_PCI_AARDVARK=m
 CONFIG_PCIE_ALTERA=m
 CONFIG_PCIE_ALTERA_MSI=m
 CONFIG_PCIE_APPLE=m
-CONFIG_PCI_VERSATILE=y
 CONFIG_PCI_HOST_THUNDER_PEM=y
 CONFIG_PCI_HOST_THUNDER_ECAM=y
 CONFIG_PCI_FTPCI100=y
 CONFIG_PCI_HOST_GENERIC=y
+CONFIG_PCIE_HISI_ERR=y
 CONFIG_PCIE_MEDIATEK=m
 CONFIG_PCIE_MEDIATEK_GEN3=m
-CONFIG_PCIE_MT7621=m
 CONFIG_PCIE_PHYTIUM_EP=m
 CONFIG_PCIE_MICROCHIP_HOST=m
+CONFIG_PCI_HYPERV_INTERFACE=m
 CONFIG_PCI_TEGRA=y
 CONFIG_PCIE_RCAR_HOST=y
 CONFIG_PCIE_RCAR_EP=y
 CONFIG_PCIE_ROCKCHIP_HOST=m
 CONFIG_PCIE_ROCKCHIP_EP=y
-CONFIG_PCI_V3_SEMI=y
 CONFIG_PCI_XGENE=y
 CONFIG_PCIE_XILINX=y
 CONFIG_PCIE_XILINX_NWL=y
@@ -777,9 +773,6 @@ CONFIG_PCIE_CADENCE_PLAT_EP=y
 CONFIG_PCI_J721E_HOST=y
 CONFIG_PCI_J721E_EP=y
 CONFIG_PCIE_AL=y
-CONFIG_PCIE_ARTPEC6_HOST=y
-CONFIG_PCIE_ARTPEC6_EP=y
-CONFIG_PCIE_BT1=m
 CONFIG_PCI_IMX6_HOST=y
 CONFIG_PCI_IMX6_EP=y
 CONFIG_PCI_LAYERSCAPE=y
@@ -787,22 +780,17 @@ CONFIG_PCI_LAYERSCAPE_EP=y
 CONFIG_PCI_HISI=y
 CONFIG_PCIE_KIRIN=y
 CONFIG_PCIE_HISI_STB=y
-CONFIG_PCIE_INTEL_GW=y
 CONFIG_PCIE_KEEMBAY_HOST=y
 CONFIG_PCIE_KEEMBAY_EP=y
 CONFIG_PCIE_ARMADA_8K=y
-CONFIG_PCIE_TEGRA194_HOST=m
-CONFIG_PCIE_TEGRA194_EP=m
 CONFIG_PCIE_DW_PLAT_HOST=y
 CONFIG_PCIE_DW_PLAT_EP=y
 CONFIG_PCIE_QCOM=y
 CONFIG_PCIE_QCOM_EP=m
 CONFIG_PCIE_ROCKCHIP_DW_HOST=y
 CONFIG_PCI_EXYNOS=m
-CONFIG_PCIE_FU740=y
 CONFIG_PCIE_UNIPHIER=y
 CONFIG_PCIE_UNIPHIER_EP=y
-CONFIG_PCIE_SPEAR13XX=y
 CONFIG_PCI_KEYSTONE_HOST=y
 CONFIG_PCI_KEYSTONE_EP=y
 CONFIG_PCIE_VISCONTI_HOST=y
@@ -816,6 +804,7 @@ CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
 # CONFIG_STANDALONE is not set
 # CONFIG_PREVENT_FIRMWARE_BUILD is not set
+# CONFIG_FW_LOADER_DEBUG is not set
 CONFIG_FW_LOADER_USER_HELPER=y
 CONFIG_FW_LOADER_USER_HELPER_FALLBACK=y
 CONFIG_FW_LOADER_COMPRESS=y
@@ -831,8 +820,6 @@ CONFIG_ISCSI_IBFT=m
 CONFIG_FW_CFG_SYSFS=m
 CONFIG_FW_CFG_SYSFS_CMDLINE=y
 CONFIG_ARM_FFA_TRANSPORT=m
-CONFIG_BCM47XX_NVRAM=y
-CONFIG_BCM47XX_SPROM=y
 CONFIG_EFI_VARS_PSTORE=m
 CONFIG_EFI_VARS_PSTORE_DEFAULT_DISABLE=y
 CONFIG_EFI_ZBOOT=y
@@ -847,16 +834,12 @@ CONFIG_GNSS_SIRF_SERIAL=m
 CONFIG_GNSS_UBX_SERIAL=m
 CONFIG_GNSS_USB=m
 CONFIG_MTD=m
-CONFIG_MTD_TESTS=m
 CONFIG_MTD_AR7_PARTS=m
-CONFIG_MTD_BCM63XX_PARTS=y
 CONFIG_MTD_BRCM_U_BOOT=m
 CONFIG_MTD_CMDLINE_PARTS=m
 CONFIG_MTD_OF_PARTS_LINKSYS_NS=y
 CONFIG_MTD_AFS_PARTS=m
-CONFIG_MTD_PARSER_TPLINK_SAFELOADER=m
 CONFIG_MTD_PARSER_TRX=m
-CONFIG_MTD_SHARPSL_PARTS=m
 CONFIG_MTD_REDBOOT_PARTS=m
 CONFIG_MTD_BLOCK=m
 CONFIG_MTD_BLOCK_RO=m
@@ -901,8 +884,33 @@ CONFIG_MTD_ONENAND_GENERIC=m
 CONFIG_MTD_ONENAND_OTP=y
 CONFIG_MTD_ONENAND_2X_PROGRAM=y
 CONFIG_MTD_RAW_NAND=m
+CONFIG_MTD_NAND_DENALI_PCI=m
+CONFIG_MTD_NAND_DENALI_DT=m
+CONFIG_MTD_NAND_CAFE=m
 CONFIG_MTD_NAND_PHYTIUM_PCI=m
 CONFIG_MTD_NAND_PHYTIUM_PLAT=m
+CONFIG_MTD_NAND_MARVELL=m
+CONFIG_MTD_NAND_BRCMNAND=m
+CONFIG_MTD_NAND_BRCMNAND_BCM63XX=m
+CONFIG_MTD_NAND_FSL_IFC=m
+CONFIG_MTD_NAND_MXC=m
+CONFIG_MTD_NAND_SUNXI=m
+CONFIG_MTD_NAND_HISI504=m
+CONFIG_MTD_NAND_QCOM=m
+CONFIG_MTD_NAND_MXIC=m
+CONFIG_MTD_NAND_TEGRA=m
+CONFIG_MTD_NAND_STM32_FMC2=m
+CONFIG_MTD_NAND_MESON=m
+CONFIG_MTD_NAND_GPIO=m
+CONFIG_MTD_NAND_PLATFORM=m
+CONFIG_MTD_NAND_CADENCE=m
+CONFIG_MTD_NAND_ARASAN=m
+CONFIG_MTD_NAND_INTEL_LGM=m
+CONFIG_MTD_NAND_ROCKCHIP=m
+CONFIG_MTD_NAND_RENESAS=m
+CONFIG_MTD_NAND_NANDSIM=m
+CONFIG_MTD_NAND_RICOH=m
+CONFIG_MTD_NAND_DISKONCHIP=m
 # CONFIG_MTD_NAND_ECC_MXIC is not set
 CONFIG_MTD_LPDDR=m
 CONFIG_MTD_SPI_NOR=m
@@ -920,7 +928,6 @@ CONFIG_PARPORT_PC_FIFO=y
 CONFIG_BLK_DEV_PCIESSD_MTIP32XX=m
 CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_DRBD=m
-CONFIG_DRBD_FAULT_INJECTION=y
 CONFIG_BLK_DEV_NBD=m
 CONFIG_BLK_DEV_RAM=y
 CONFIG_BLK_DEV_RAM_COUNT=8
@@ -952,13 +959,9 @@ CONFIG_DUMMY_IRQ=m
 CONFIG_PHANTOM=m
 CONFIG_PHYTIUM_LPC_SNOOP=m
 CONFIG_ICS932S401=m
-CONFIG_ATMEL_SSC=m
 CONFIG_ENCLOSURE_SERVICES=m
-CONFIG_SMPRO_ERRMON=m
-CONFIG_SMPRO_MISC=m
-CONFIG_GEHC_ACHC=m
+CONFIG_HI6421V600_IRQ=m
 CONFIG_HP_ILO=m
-CONFIG_QCOM_COINCELL=m
 CONFIG_QCOM_FASTRPC=m
 CONFIG_APDS9802ALS=m
 CONFIG_ISL29003=m
@@ -968,11 +971,8 @@ CONFIG_SENSORS_BH1770=m
 CONFIG_SENSORS_APDS990X=m
 CONFIG_HMC6352=m
 CONFIG_DS1682=m
-CONFIG_PCH_PHUB=m
 CONFIG_LATTICE_ECP3_CONFIG=m
-CONFIG_SRAM=y
 CONFIG_DW_XDATA_PCIE=m
-CONFIG_PCI_ENDPOINT_TEST=m
 CONFIG_XILINX_SDFEC=m
 CONFIG_HISI_HIKEY_USB=m
 CONFIG_OPEN_DICE=m
@@ -1073,7 +1073,6 @@ CONFIG_SCSI_QLA_ISCSI=m
 CONFIG_QEDI=m
 CONFIG_QEDF=m
 CONFIG_SCSI_LPFC=m
-CONFIG_SCSI_LPFC_DEBUG_FS=y
 CONFIG_SCSI_DC395x=m
 CONFIG_SCSI_AM53C974=m
 CONFIG_SCSI_WD719X=m
@@ -1092,20 +1091,14 @@ CONFIG_SATA_ZPODD=y
 CONFIG_SATA_AHCI=y
 CONFIG_SATA_AHCI_PLATFORM=m
 CONFIG_AHCI_BRCM=m
-CONFIG_AHCI_DA850=m
-CONFIG_AHCI_DM816=m
 CONFIG_AHCI_DWC=m
-CONFIG_AHCI_ST=m
 CONFIG_AHCI_IMX=m
 CONFIG_AHCI_CEVA=m
 CONFIG_AHCI_MTK=m
 CONFIG_AHCI_MVEBU=m
 CONFIG_AHCI_SUNXI=m
 CONFIG_AHCI_TEGRA=m
-CONFIG_AHCI_XGENE=m
 CONFIG_AHCI_QORIQ=m
-CONFIG_SATA_FSL=m
-CONFIG_SATA_GEMINI=m
 CONFIG_SATA_AHCI_SEATTLE=m
 CONFIG_SATA_INIC162X=m
 CONFIG_SATA_ACARD_AHCI=m
@@ -1116,7 +1109,6 @@ CONFIG_SATA_SX4=m
 CONFIG_ATA_PIIX=m
 CONFIG_SATA_DWC=m
 CONFIG_SATA_DWC_OLD_DMA=y
-CONFIG_SATA_HIGHBANK=m
 CONFIG_SATA_MV=m
 CONFIG_SATA_NV=m
 CONFIG_SATA_PROMISE=m
@@ -1129,17 +1121,12 @@ CONFIG_SATA_VIA=m
 CONFIG_SATA_VITESSE=m
 CONFIG_PATA_ALI=m
 CONFIG_PATA_AMD=m
-CONFIG_PATA_ARASAN_CF=m
 CONFIG_PATA_ARTOP=m
 CONFIG_PATA_ATIIXP=m
 CONFIG_PATA_ATP867X=m
 CONFIG_PATA_CMD64X=m
-CONFIG_PATA_CS5520=m
-CONFIG_PATA_CS5530=m
-CONFIG_PATA_CS5536=m
 CONFIG_PATA_CYPRESS=m
 CONFIG_PATA_EFAR=m
-CONFIG_PATA_FTIDE010=m
 CONFIG_PATA_HPT366=m
 CONFIG_PATA_HPT37X=m
 CONFIG_PATA_HPT3X2N=m
@@ -1159,17 +1146,14 @@ CONFIG_PATA_PDC2027X=m
 CONFIG_PATA_PDC_OLD=m
 CONFIG_PATA_RADISYS=m
 CONFIG_PATA_RDC=m
-CONFIG_PATA_SC1200=m
 CONFIG_PATA_SCH=m
 CONFIG_PATA_SERVERWORKS=m
 CONFIG_PATA_SIL680=m
 CONFIG_PATA_TOSHIBA=m
 CONFIG_PATA_TRIFLEX=m
 CONFIG_PATA_VIA=m
-CONFIG_PATA_PXA=m
 CONFIG_PATA_WINBOND=m
 CONFIG_PATA_CMD640_PCI=m
-CONFIG_PATA_IXP4XX_CF=m
 CONFIG_PATA_MPIIX=m
 CONFIG_PATA_NS87410=m
 CONFIG_PATA_OPTI=m
@@ -1202,7 +1186,6 @@ CONFIG_MD_FAULTY=m
 CONFIG_MD_CLUSTER=m
 CONFIG_BCACHE=m
 CONFIG_BLK_DEV_DM=m
-CONFIG_DM_DEBUG_BLOCK_MANAGER_LOCKING=y
 CONFIG_DM_UNSTRIPED=m
 CONFIG_DM_CRYPT=m
 CONFIG_DM_SNAPSHOT=m
@@ -1452,7 +1435,10 @@ CONFIG_SMSC911X=m
 CONFIG_SMSC9420=m
 CONFIG_STMMAC_ETH=y
 CONFIG_DWMAC_DWC_QOS_ETH=m
+CONFIG_DWMAC_MEDIATEK=m
 CONFIG_DWMAC_PHYTIUM=m
+CONFIG_DWMAC_INTEL_PLAT=m
+CONFIG_DWMAC_TEGRA=m
 CONFIG_STMMAC_PCI=y
 CONFIG_HAPPYMEAL=m
 CONFIG_SUNGEM=m
@@ -1616,8 +1602,6 @@ CONFIG_WCN36XX=m
 CONFIG_ATH11K=m
 CONFIG_ATH11K_PCI=m
 CONFIG_ATH12K=m
-CONFIG_ATH12K_DEBUG=y
-CONFIG_ATH12K_TRACING=y
 CONFIG_ATMEL=m
 CONFIG_PCI_ATMEL=m
 CONFIG_AT76C50X_USB=m
@@ -1777,6 +1761,7 @@ CONFIG_INPUT_MOUSEDEV_PSAUX=y
 CONFIG_INPUT_EVDEV=y
 CONFIG_KEYBOARD_ADP5588=m
 CONFIG_KEYBOARD_ADP5589=m
+CONFIG_KEYBOARD_QT1050=m
 CONFIG_KEYBOARD_QT1070=m
 CONFIG_KEYBOARD_QT2160=m
 CONFIG_KEYBOARD_DLINK_DIR685=m
@@ -1805,6 +1790,7 @@ CONFIG_KEYBOARD_XTKBD=m
 CONFIG_KEYBOARD_CAP11XX=m
 CONFIG_KEYBOARD_BCM=m
 CONFIG_KEYBOARD_MTK_PMIC=m
+CONFIG_KEYBOARD_CYPRESS_SF=m
 CONFIG_KEYBOARD_PHYTIUM=m
 CONFIG_MOUSE_PS2_SENTELIC=y
 CONFIG_MOUSE_PS2_TOUCHKIT=y
@@ -1891,8 +1877,6 @@ CONFIG_TOUCHSCREEN_IMX6UL_TSC=m
 CONFIG_TOUCHSCREEN_INEXIO=m
 CONFIG_TOUCHSCREEN_PENMOUNT=m
 CONFIG_TOUCHSCREEN_EDT_FT5X06=m
-CONFIG_TOUCHSCREEN_RASPBERRYPI_FW=m
-CONFIG_TOUCHSCREEN_MIGOR=m
 CONFIG_TOUCHSCREEN_TOUCHRIGHT=m
 CONFIG_TOUCHSCREEN_TOUCHWIN=m
 CONFIG_TOUCHSCREEN_PIXCIR=m
@@ -1902,7 +1886,6 @@ CONFIG_TOUCHSCREEN_WM97XX=m
 CONFIG_TOUCHSCREEN_USB_COMPOSITE=m
 CONFIG_TOUCHSCREEN_MC13783=m
 CONFIG_TOUCHSCREEN_TOUCHIT213=m
-CONFIG_TOUCHSCREEN_TS4800=m
 CONFIG_TOUCHSCREEN_TSC_SERIO=m
 CONFIG_TOUCHSCREEN_TSC2004=m
 CONFIG_TOUCHSCREEN_TSC2005=m
@@ -1988,7 +1971,6 @@ CONFIG_SERIO_ALTERA_PS2=m
 CONFIG_SERIO_PS2MULT=m
 CONFIG_SERIO_ARC_PS2=m
 CONFIG_SERIO_APBPS2=m
-CONFIG_SERIO_OLPC_APSP=m
 CONFIG_SERIO_SUN4I_PS2=m
 CONFIG_SERIO_GPIO_PS2=m
 CONFIG_USERIO=m
@@ -2000,33 +1982,27 @@ CONFIG_SERIAL_8250_FINTEK=y
 CONFIG_SERIAL_8250_CONSOLE=y
 CONFIG_SERIAL_8250_NR_UARTS=48
 CONFIG_SERIAL_8250_RUNTIME_UARTS=32
+CONFIG_SERIAL_8250_EXTENDED=y
 CONFIG_SERIAL_8250_MANY_PORTS=y
-CONFIG_SERIAL_8250_ASPEED_VUART=m
 CONFIG_SERIAL_8250_PCI1XXXX=m
+CONFIG_SERIAL_8250_SHARE_IRQ=y
 CONFIG_SERIAL_8250_DETECT_IRQ=y
 CONFIG_SERIAL_8250_RSA=y
 CONFIG_SERIAL_8250_BCM2835AUX=m
 CONFIG_SERIAL_8250_DW=y
 CONFIG_SERIAL_8250_EM=m
-CONFIG_SERIAL_8250_IOC3=m
 CONFIG_SERIAL_8250_RT288X=y
 CONFIG_SERIAL_8250_OMAP=m
-CONFIG_SERIAL_8250_LPC18XX=m
 CONFIG_SERIAL_8250_MT6577=m
 CONFIG_SERIAL_8250_UNIPHIER=m
-CONFIG_SERIAL_8250_INGENIC=m
-CONFIG_SERIAL_8250_PXA=m
 CONFIG_SERIAL_OF_PLATFORM=y
 CONFIG_SERIAL_AMBA_PL010=y
 CONFIG_SERIAL_AMBA_PL010_CONSOLE=y
 CONFIG_SERIAL_AMBA_PL011=y
 CONFIG_SERIAL_AMBA_PL011_CONSOLE=y
 CONFIG_SERIAL_PHYTIUM_PCI=m
-CONFIG_SERIAL_ATMEL=y
-CONFIG_SERIAL_ATMEL_CONSOLE=y
 CONFIG_SERIAL_MESON=m
 CONFIG_SERIAL_MESON_CONSOLE=y
-CONFIG_SERIAL_CLPS711X=m
 CONFIG_SERIAL_SAMSUNG=m
 CONFIG_SERIAL_SAMSUNG_CONSOLE=y
 CONFIG_SERIAL_TEGRA=m
@@ -2037,43 +2013,26 @@ CONFIG_SERIAL_IMX=m
 CONFIG_SERIAL_IMX_CONSOLE=m
 CONFIG_SERIAL_UARTLITE=m
 CONFIG_SERIAL_SH_SCI=m
-CONFIG_SERIAL_HS_LPC32XX=m
-CONFIG_SERIAL_ICOM=m
 CONFIG_SERIAL_JSM=m
 CONFIG_SERIAL_MSM=m
-CONFIG_SERIAL_VT8500=y
-CONFIG_SERIAL_VT8500_CONSOLE=y
-CONFIG_SERIAL_OMAP=m
 CONFIG_SERIAL_SIFIVE=m
-CONFIG_SERIAL_LANTIQ=m
 CONFIG_SERIAL_SCCNXP=m
 CONFIG_SERIAL_SC16IS7XX=m
 CONFIG_SERIAL_SC16IS7XX_SPI=y
-CONFIG_SERIAL_TIMBERDALE=m
 CONFIG_SERIAL_BCM63XX_CONSOLE=y
 CONFIG_SERIAL_ALTERA_JTAGUART=m
 CONFIG_SERIAL_ALTERA_UART=m
-CONFIG_SERIAL_PCH_UART=m
-CONFIG_SERIAL_MXS_AUART=m
 CONFIG_SERIAL_XILINX_PS_UART=y
 CONFIG_SERIAL_XILINX_PS_UART_CONSOLE=y
-CONFIG_SERIAL_MPS2_UART_CONSOLE=y
-CONFIG_SERIAL_MPS2_UART=y
 CONFIG_SERIAL_ARC=m
 CONFIG_SERIAL_RP2=m
 CONFIG_SERIAL_FSL_LPUART=m
 CONFIG_SERIAL_FSL_LINFLEXUART=m
 CONFIG_SERIAL_CONEXANT_DIGICOLOR=m
-CONFIG_SERIAL_ST_ASC=m
 CONFIG_SERIAL_SPRD=m
 CONFIG_SERIAL_STM32=m
 CONFIG_SERIAL_MVEBU_UART=y
 CONFIG_SERIAL_OWL=m
-CONFIG_SERIAL_RDA=y
-CONFIG_SERIAL_MILBEAUT_USIO=m
-CONFIG_SERIAL_LITEUART=m
-CONFIG_SERIAL_SUNPLUS=m
-CONFIG_SERIAL_SUNPLUS_CONSOLE=y
 CONFIG_SERIAL_NUVOTON_MA35D1=m
 CONFIG_SERIAL_NONSTANDARD=y
 CONFIG_MOXA_INTELLIO=m
@@ -2084,20 +2043,27 @@ CONFIG_NOZOMI=m
 CONFIG_HVC_DCC=y
 CONFIG_RPMSG_TTY=m
 CONFIG_SERIAL_DEV_BUS=m
+CONFIG_TTY_PRINTK=m
+CONFIG_TTY_PRINTK_LEVEL=7
+CONFIG_PRINTER=m
+CONFIG_LP_CONSOLE=y
+CONFIG_PPDEV=m
 CONFIG_VIRTIO_CONSOLE=y
-CONFIG_ASPEED_KCS_IPMI_BMC=m
 CONFIG_PHYTIUM_KCS_IPMI_BMC=m
 CONFIG_PHYTIUM_BT_IPMI_BMC=m
 CONFIG_NPCM7XX_KCS_IPMI_BMC=m
 CONFIG_IPMI_KCS_BMC_CDEV_IPMI=m
 CONFIG_IPMI_KCS_BMC_SERIO=m
-CONFIG_ASPEED_BT_IPMI_BMC=m
 CONFIG_SSIF_IPMI_BMC=m
+CONFIG_IPMB_DEVICE_INTERFACE=m
 CONFIG_HW_RANDOM=y
 CONFIG_HW_RANDOM_TIMERIOMEM=m
 CONFIG_HW_RANDOM_VIRTIO=m
 CONFIG_HW_RANDOM_HISI=m
+CONFIG_HW_RANDOM_CCTRNG=m
+CONFIG_HW_RANDOM_XIPHERA=m
 CONFIG_HW_RANDOM_PHYTIUM=m
+CONFIG_APPLICOM=m
 CONFIG_TCG_TIS=m
 CONFIG_TCG_TIS_SPI=m
 CONFIG_TCG_TIS_SPI_CR50=y
@@ -2127,7 +2093,6 @@ CONFIG_I2C_MUX_PINCTRL=m
 CONFIG_I2C_MUX_REG=m
 CONFIG_I2C_DEMUX_PINCTRL=m
 CONFIG_I2C_MUX_MLXCPLD=m
-CONFIG_I2C_ATR=m
 CONFIG_I2C_ALI1535=m
 CONFIG_I2C_ALI1563=m
 CONFIG_I2C_ALI15X3=m
@@ -2147,35 +2112,19 @@ CONFIG_I2C_VIA=m
 CONFIG_I2C_VIAPRO=m
 CONFIG_I2C_SCMI=m
 CONFIG_I2C_ALTERA=m
-CONFIG_I2C_ASPEED=m
-CONFIG_I2C_AT91=m
-CONFIG_I2C_AT91_SLAVE_EXPERIMENTAL=m
-CONFIG_I2C_AXXIA=m
 CONFIG_I2C_BCM2835=m
-CONFIG_I2C_BCM_KONA=m
 CONFIG_I2C_CADENCE=m
 CONFIG_I2C_CBUS_GPIO=m
-CONFIG_I2C_DAVINCI=m
 CONFIG_I2C_DESIGNWARE_SLAVE=y
 CONFIG_I2C_DESIGNWARE_PCI=m
-CONFIG_I2C_DIGICOLOR=m
-CONFIG_I2C_EG20T=m
 CONFIG_I2C_EMEV2=m
 CONFIG_I2C_GPIO=m
-CONFIG_I2C_GXP=m
-CONFIG_I2C_IMG=m
+CONFIG_I2C_HISI=m
 CONFIG_I2C_IMX=m
 CONFIG_I2C_IMX_LPI2C=m
-CONFIG_I2C_IOP3XX=m
-CONFIG_I2C_JZ4780=m
-CONFIG_I2C_LPC2K=m
-CONFIG_I2C_LS2X=m
 CONFIG_I2C_MESON=m
-CONFIG_I2C_MICROCHIP_CORE=m
 CONFIG_I2C_MT65XX=m
-CONFIG_I2C_MT7621=m
 CONFIG_I2C_MV64XXX=m
-CONFIG_I2C_MXS=m
 CONFIG_I2C_NOMADIK=m
 CONFIG_I2C_NPCM=m
 CONFIG_I2C_OCORES=m
@@ -2184,7 +2133,6 @@ CONFIG_I2C_OWL=m
 CONFIG_I2C_PCA_PLATFORM=m
 CONFIG_I2C_PHYTIUM_PCI=m
 CONFIG_I2C_PHYTIUM_PLATFORM=m
-CONFIG_I2C_PNX=m
 CONFIG_I2C_PXA=m
 CONFIG_I2C_PXA_SLAVE=y
 CONFIG_I2C_QCOM_CCI=m
@@ -2196,17 +2144,13 @@ CONFIG_I2C_S3C2410=m
 CONFIG_I2C_SH_MOBILE=m
 CONFIG_I2C_SIMTEC=m
 CONFIG_I2C_SPRD=m
-CONFIG_I2C_ST=m
 CONFIG_I2C_STM32F4=m
 CONFIG_I2C_STM32F7=m
-CONFIG_I2C_SUN6I_P2WI=m
 CONFIG_I2C_SYNQUACER=m
 CONFIG_I2C_TEGRA=m
-CONFIG_I2C_TEGRA_BPMP=m
 CONFIG_I2C_UNIPHIER=m
 CONFIG_I2C_UNIPHIER_F=m
 CONFIG_I2C_VERSATILE=m
-CONFIG_I2C_WMT=m
 CONFIG_I2C_THUNDERX=m
 CONFIG_I2C_XILINX=m
 CONFIG_I2C_XLP9XX=m
@@ -2225,23 +2169,15 @@ CONFIG_I2C_XGENE_SLIMPRO=m
 CONFIG_I2C_VIRTIO=m
 CONFIG_I2C_STUB=m
 CONFIG_I2C_SLAVE_EEPROM=m
-CONFIG_I2C_SLAVE_TESTUNIT=m
 CONFIG_I3C=m
 CONFIG_PHYTIUM_I3C_MASTER=m
 CONFIG_SPI=y
 CONFIG_SPI_ALTERA=m
 CONFIG_SPI_AMLOGIC_SPIFC_A1=m
-CONFIG_SPI_AR934X=m
-CONFIG_SPI_ATH79=m
 CONFIG_SPI_ARMADA_3700=m
-CONFIG_SPI_ASPEED_SMC=m
-CONFIG_SPI_ATMEL=m
-CONFIG_SPI_AT91_USART=m
-CONFIG_SPI_ATMEL_QUADSPI=m
 CONFIG_SPI_AXI_SPI_ENGINE=m
 CONFIG_SPI_BCM2835=m
 CONFIG_SPI_BCM2835AUX=m
-CONFIG_SPI_BCM63XX=m
 CONFIG_SPI_BCM63XX_HSSPI=m
 CONFIG_SPI_BCM_QSPI=m
 CONFIG_SPI_BCMBCA_HSSPI=m
@@ -2249,29 +2185,17 @@ CONFIG_SPI_BUTTERFLY=m
 CONFIG_SPI_CADENCE=m
 CONFIG_SPI_CADENCE_QUADSPI=m
 CONFIG_SPI_CADENCE_XSPI=m
-CONFIG_SPI_CLPS711X=m
-CONFIG_SPI_DAVINCI=m
 CONFIG_SPI_DESIGNWARE=m
 CONFIG_SPI_DW_DMA=y
 CONFIG_SPI_DW_PCI=m
 CONFIG_SPI_DW_MMIO=m
-CONFIG_SPI_DW_BT1=m
-CONFIG_SPI_DW_BT1_DIRMAP=y
 CONFIG_SPI_DLN2=m
-CONFIG_SPI_EP93XX=m
 CONFIG_SPI_FSL_LPSPI=m
 CONFIG_SPI_FSL_QUADSPI=m
-CONFIG_SPI_GXP=m
 CONFIG_SPI_NXP_FLEXSPI=m
 CONFIG_SPI_GPIO=m
-CONFIG_SPI_IMG_SPFI=m
 CONFIG_SPI_IMX=m
-CONFIG_SPI_INGENIC=m
-CONFIG_SPI_INTEL_PCI=m
-CONFIG_SPI_INTEL_PLATFORM=m
-CONFIG_SPI_JCORE=m
 CONFIG_SPI_LM70_LLP=m
-CONFIG_SPI_LP8841_RTC=m
 CONFIG_SPI_FSL_SPI=m
 CONFIG_SPI_FSL_DSPI=m
 CONFIG_SPI_MESON_SPICC=m
@@ -2279,19 +2203,17 @@ CONFIG_SPI_MESON_SPIFC=m
 CONFIG_SPI_MICROCHIP_CORE=m
 CONFIG_SPI_MICROCHIP_CORE_QSPI=m
 CONFIG_SPI_MT65XX=m
-CONFIG_SPI_MT7621=m
 CONFIG_SPI_MTK_NOR=m
 CONFIG_SPI_WPCM_FIU=m
 CONFIG_SPI_NPCM_FIU=m
 CONFIG_SPI_NPCM_PSPI=m
-CONFIG_SPI_LANTIQ_SSC=m
 CONFIG_SPI_OC_TINY=m
 CONFIG_SPI_OMAP24XX=m
-CONFIG_SPI_TI_QSPI=m
 CONFIG_SPI_ORION=y
 CONFIG_SPI_PCI1XXXX=m
-CONFIG_SPI_PIC32=m
-CONFIG_SPI_PIC32_SQI=m
+CONFIG_SPI_PHYTIUM_PLAT=m
+CONFIG_SPI_PHYTIUM_PCI=m
+CONFIG_SPI_PHYTIUM_QSPI=m
 CONFIG_SPI_PL022=y
 CONFIG_SPI_PXA2XX=m
 CONFIG_SPI_ROCKCHIP=m
@@ -2303,7 +2225,6 @@ CONFIG_SPI_QUP=m
 CONFIG_SPI_S3C64XX=m
 CONFIG_SPI_SC18IS602=m
 CONFIG_SPI_SH_MSIOF=m
-CONFIG_SPI_SH=m
 CONFIG_SPI_SH_HSPI=m
 CONFIG_SPI_SIFIVE=m
 CONFIG_SPI_SN_F_OSPI=m
@@ -2311,10 +2232,8 @@ CONFIG_SPI_SPRD=m
 CONFIG_SPI_SPRD_ADI=m
 CONFIG_SPI_STM32=m
 CONFIG_SPI_STM32_QSPI=m
-CONFIG_SPI_ST_SSC4=m
 CONFIG_SPI_SUN4I=m
 CONFIG_SPI_SUN6I=m
-CONFIG_SPI_SUNPLUS_SP7021=m
 CONFIG_SPI_SYNQUACER=m
 CONFIG_SPI_MXIC=m
 CONFIG_SPI_TEGRA210_QUAD=m
@@ -2322,13 +2241,10 @@ CONFIG_SPI_TEGRA114=m
 CONFIG_SPI_TEGRA20_SFLASH=m
 CONFIG_SPI_TEGRA20_SLINK=m
 CONFIG_SPI_THUNDERX=m
-CONFIG_SPI_TOPCLIFF_PCH=m
 CONFIG_SPI_UNIPHIER=m
 CONFIG_SPI_XCOMM=m
 CONFIG_SPI_XILINX=m
 CONFIG_SPI_XLP=m
-CONFIG_SPI_XTENSA_XTFPGA=m
-CONFIG_SPI_ZYNQ_QSPI=m
 CONFIG_SPI_ZYNQMP_GQSPI=m
 CONFIG_SPI_AMD=m
 CONFIG_SPI_MUX=m
@@ -2353,20 +2269,22 @@ CONFIG_GPIO_PCA953X=m
 CONFIG_GPIO_PCF857X=m
 CONFIG_GPIO_DLN2=m
 CONFIG_GPIO_LP3943=m
-CONFIG_GPIO_PHYTIUM_PCI=m
-CONFIG_GPIO_AMD8111=m
 CONFIG_GPIO_BT8XX=m
 CONFIG_GPIO_MLXBF=m
 CONFIG_GPIO_MLXBF2=m
 CONFIG_GPIO_MLXBF3=m
-CONFIG_GPIO_ML_IOH=m
-CONFIG_GPIO_PCH=m
 CONFIG_GPIO_PCI_IDIO_16=m
 CONFIG_GPIO_PCIE_IDIO_24=m
 CONFIG_GPIO_RDC321X=m
 CONFIG_GPIO_VIPERBOARD=m
 CONFIG_W1=m
+CONFIG_W1_MASTER_MATROX=m
+CONFIG_W1_MASTER_DS2490=m
+CONFIG_W1_MASTER_DS2482=m
+CONFIG_W1_MASTER_MXC=m
+CONFIG_W1_MASTER_GPIO=m
 CONFIG_W1_MASTER_PHYTIUM=y
+CONFIG_W1_MASTER_SGI=m
 # CONFIG_POWER_RESET_BRCMSTB is not set
 CONFIG_POWER_RESET_GPIO=y
 CONFIG_POWER_RESET_GPIO_RESTART=y
@@ -2405,41 +2323,76 @@ CONFIG_SENSORS_ADT7411=m
 CONFIG_SENSORS_ADT7462=m
 CONFIG_SENSORS_ADT7470=m
 CONFIG_SENSORS_ADT7475=m
+CONFIG_SENSORS_AHT10=m
+CONFIG_SENSORS_AQUACOMPUTER_D5NEXT=m
+CONFIG_SENSORS_AS370=m
 CONFIG_SENSORS_ASC7621=m
+CONFIG_SENSORS_AXI_FAN_CONTROL=m
 CONFIG_SENSORS_ARM_SCMI=m
 CONFIG_SENSORS_ARM_SCPI=m
 CONFIG_SENSORS_PHYTIUM=m
 CONFIG_SENSORS_ATXP1=m
+CONFIG_SENSORS_CORSAIR_CPRO=m
+CONFIG_SENSORS_CORSAIR_PSU=m
+CONFIG_SENSORS_DRIVETEMP=m
 CONFIG_SENSORS_DS620=m
 CONFIG_SENSORS_DS1621=m
+CONFIG_SENSORS_DA9052_ADC=m
+CONFIG_SENSORS_DA9055=m
 CONFIG_SENSORS_I5K_AMB=m
+CONFIG_SENSORS_SPARX5=m
 CONFIG_SENSORS_F71805F=m
 CONFIG_SENSORS_F71882FG=m
 CONFIG_SENSORS_F75375S=m
 CONFIG_SENSORS_MC13783_ADC=m
+CONFIG_SENSORS_FTSTEUTATES=m
 CONFIG_SENSORS_GL518SM=m
 CONFIG_SENSORS_GL520SM=m
 CONFIG_SENSORS_G760A=m
 CONFIG_SENSORS_G762=m
 CONFIG_SENSORS_GPIO_FAN=m
 CONFIG_SENSORS_HIH6130=m
+CONFIG_SENSORS_HS3001=m
+CONFIG_SENSORS_IIO_HWMON=m
 CONFIG_SENSORS_IT87=m
 CONFIG_SENSORS_JC42=m
 CONFIG_SENSORS_POWR1220=m
 CONFIG_SENSORS_LINEAGE=m
 CONFIG_SENSORS_LTC2945=m
+CONFIG_SENSORS_LTC2947_I2C=m
+CONFIG_SENSORS_LTC2947_SPI=m
 CONFIG_SENSORS_LTC2990=m
+CONFIG_SENSORS_LTC2992=m
 CONFIG_SENSORS_LTC4151=m
 CONFIG_SENSORS_LTC4215=m
 CONFIG_SENSORS_LTC4222=m
 CONFIG_SENSORS_LTC4245=m
 CONFIG_SENSORS_LTC4260=m
+CONFIG_SENSORS_LTC4261=m
+CONFIG_SENSORS_MAX1111=m
+CONFIG_SENSORS_MAX127=m
+CONFIG_SENSORS_MAX16065=m
 CONFIG_SENSORS_MAX1619=m
 CONFIG_SENSORS_MAX1668=m
+CONFIG_SENSORS_MAX197=m
+CONFIG_SENSORS_MAX31722=m
+CONFIG_SENSORS_MAX31730=m
+CONFIG_SENSORS_MAX31760=m
+CONFIG_MAX31827=m
+CONFIG_SENSORS_MAX6620=m
+CONFIG_SENSORS_MAX6621=m
 CONFIG_SENSORS_MAX6639=m
 CONFIG_SENSORS_MAX6650=m
 CONFIG_SENSORS_MAX6697=m
 CONFIG_SENSORS_MAX31790=m
+CONFIG_SENSORS_MC34VR500=m
+CONFIG_SENSORS_MCP3021=m
+CONFIG_SENSORS_MLXREG_FAN=m
+CONFIG_SENSORS_TC654=m
+CONFIG_SENSORS_TPS23861=m
+CONFIG_SENSORS_MENF21BMC_HWMON=m
+CONFIG_SENSORS_MR75203=m
+CONFIG_SENSORS_ADCXX=m
 CONFIG_SENSORS_LM63=m
 CONFIG_SENSORS_LM70=m
 CONFIG_SENSORS_LM73=m
@@ -2458,11 +2411,16 @@ CONFIG_SENSORS_LM95241=m
 CONFIG_SENSORS_LM95245=m
 CONFIG_SENSORS_PC87360=m
 CONFIG_SENSORS_PC87427=m
+CONFIG_SENSORS_NTC_THERMISTOR=m
 CONFIG_SENSORS_NCT6683=m
 CONFIG_SENSORS_NCT6775=m
+CONFIG_SENSORS_NCT6775_I2C=m
 CONFIG_SENSORS_NCT7802=m
 CONFIG_SENSORS_NCT7904=m
 CONFIG_SENSORS_NPCM7XX=m
+CONFIG_SENSORS_NZXT_KRAKEN2=m
+CONFIG_SENSORS_NZXT_SMART2=m
+CONFIG_SENSORS_OCC_P8_I2C=m
 CONFIG_SENSORS_PCF8591=m
 CONFIG_PMBUS=m
 CONFIG_SENSORS_ADM1275=m
@@ -2477,43 +2435,61 @@ CONFIG_SENSORS_UCD9000=m
 CONFIG_SENSORS_UCD9200=m
 CONFIG_SENSORS_ZL6100=m
 CONFIG_SENSORS_PWM_FAN=m
+CONFIG_SENSORS_SBTSI=m
+CONFIG_SENSORS_SBRMI=m
 CONFIG_SENSORS_SHT15=m
 CONFIG_SENSORS_SHT21=m
 CONFIG_SENSORS_SHT3x=m
+CONFIG_SENSORS_SHT4x=m
 CONFIG_SENSORS_SHTC1=m
 CONFIG_SENSORS_SIS5595=m
 CONFIG_SENSORS_DME1737=m
 CONFIG_SENSORS_EMC1403=m
 CONFIG_SENSORS_EMC2103=m
+CONFIG_SENSORS_EMC2305=m
 CONFIG_SENSORS_EMC6W201=m
 CONFIG_SENSORS_SMSC47M1=m
 CONFIG_SENSORS_SMSC47M192=m
 CONFIG_SENSORS_SMSC47B397=m
+CONFIG_SENSORS_SCH5627=m
+CONFIG_SENSORS_SCH5636=m
+CONFIG_SENSORS_STTS751=m
 CONFIG_SENSORS_ADC128D818=m
 CONFIG_SENSORS_ADS7828=m
+CONFIG_SENSORS_ADS7871=m
 CONFIG_SENSORS_AMC6821=m
 CONFIG_SENSORS_INA209=m
 CONFIG_SENSORS_INA2XX=m
+CONFIG_SENSORS_INA238=m
+CONFIG_SENSORS_INA3221=m
 CONFIG_SENSORS_TC74=m
 CONFIG_SENSORS_THMC50=m
 CONFIG_SENSORS_TMP102=m
 CONFIG_SENSORS_TMP103=m
+CONFIG_SENSORS_TMP108=m
 CONFIG_SENSORS_TMP401=m
 CONFIG_SENSORS_TMP421=m
+CONFIG_SENSORS_TMP464=m
+CONFIG_SENSORS_TMP513=m
 CONFIG_SENSORS_VEXPRESS=m
 CONFIG_SENSORS_VIA686A=m
 CONFIG_SENSORS_VT1211=m
 CONFIG_SENSORS_VT8231=m
+CONFIG_SENSORS_W83773G=m
 CONFIG_SENSORS_W83781D=m
 CONFIG_SENSORS_W83791D=m
 CONFIG_SENSORS_W83792D=m
 CONFIG_SENSORS_W83793=m
 CONFIG_SENSORS_W83795=m
+CONFIG_SENSORS_W83795_FANCTRL=y
 CONFIG_SENSORS_W83L785TS=m
 CONFIG_SENSORS_W83L786NG=m
 CONFIG_SENSORS_W83627HF=m
 CONFIG_SENSORS_W83627EHF=m
+CONFIG_SENSORS_WM831X=m
+CONFIG_SENSORS_WM8350=m
 CONFIG_SENSORS_XGENE=m
+CONFIG_SENSORS_ACPI_POWER=m
 CONFIG_THERMAL_GOV_FAIR_SHARE=y
 CONFIG_THERMAL_GOV_BANG_BANG=y
 CONFIG_THERMAL_GOV_USER_SPACE=y
@@ -2559,7 +2535,6 @@ CONFIG_MFD_RETU=m
 CONFIG_MFD_PCF50633=m
 CONFIG_PCF50633_ADC=m
 CONFIG_PCF50633_GPIO=m
-CONFIG_MFD_RDC321X=m
 CONFIG_MFD_PHYTIUM_I2S_LSD=y
 CONFIG_MFD_PHYTIUM_I2S_MMD=y
 CONFIG_MFD_RT5033=m
@@ -2770,20 +2745,66 @@ CONFIG_USB_SI4713=m
 CONFIG_PLATFORM_SI4713=m
 CONFIG_RADIO_WL128X=m
 CONFIG_V4L_PLATFORM_DRIVERS=y
+CONFIG_SDR_PLATFORM_DRIVERS=y
 CONFIG_DVB_PLATFORM_DRIVERS=y
 CONFIG_V4L_MEM2MEM_DRIVERS=y
 CONFIG_VIDEO_MEM2MEM_DEINTERLACE=m
 CONFIG_VIDEO_MUX=m
+CONFIG_VIDEO_ALLEGRO_DVT=m
+CONFIG_VIDEO_MESON_GE2D=m
+CONFIG_VIDEO_AMPHION_VPU=m
 CONFIG_VIDEO_CADENCE_CSI2RX=m
 CONFIG_VIDEO_CADENCE_CSI2TX=m
+CONFIG_VIDEO_CODA=m
 CONFIG_VIDEO_CAFE_CCIC=m
+CONFIG_VIDEO_MEDIATEK_VPU=m
+CONFIG_VIDEO_TEGRA_VDE=m
+CONFIG_VIDEO_IMX7_CSI=m
+CONFIG_VIDEO_IMX8MQ_MIPI_CSI2=m
+CONFIG_VIDEO_IMX_MIPI_CSIS=m
+CONFIG_VIDEO_IMX8_ISI=m
+CONFIG_VIDEO_IMX8_ISI_M2M=y
+CONFIG_VIDEO_IMX_PXP=m
+CONFIG_VIDEO_DW100=m
+CONFIG_VIDEO_IMX8_JPEG=m
 CONFIG_VIDEO_PHYTIUM_JPEG=m
+CONFIG_VIDEO_QCOM_CAMSS=m
+CONFIG_VIDEO_RCAR_ISP=m
+CONFIG_VIDEO_RCAR_CSI2=m
+CONFIG_VIDEO_RCAR_VIN=m
+CONFIG_VIDEO_RZG2L_CSI2=m
+CONFIG_VIDEO_RZG2L_CRU=m
+CONFIG_VIDEO_RENESAS_FCP=m
+CONFIG_VIDEO_RENESAS_FDP1=m
+CONFIG_VIDEO_RENESAS_JPU=m
+CONFIG_VIDEO_RENESAS_VSP1=m
+CONFIG_VIDEO_RCAR_DRIF=m
+CONFIG_VIDEO_ROCKCHIP_RGA=m
+CONFIG_VIDEO_ROCKCHIP_ISP1=m
+CONFIG_VIDEO_SAMSUNG_EXYNOS_GSC=m
+CONFIG_VIDEO_SAMSUNG_EXYNOS4_IS=m
+CONFIG_VIDEO_S5P_FIMC=m
+CONFIG_VIDEO_S5P_MIPI_CSIS=m
+CONFIG_VIDEO_EXYNOS4_FIMC_IS=m
+CONFIG_VIDEO_SAMSUNG_S5P_G2D=m
+CONFIG_VIDEO_SAMSUNG_S5P_JPEG=m
+CONFIG_VIDEO_SAMSUNG_S5P_MFC=m
+CONFIG_VIDEO_STM32_DCMI=m
+CONFIG_VIDEO_STM32_DMA2D=m
+CONFIG_VIDEO_SUN4I_CSI=m
+CONFIG_VIDEO_SUN6I_CSI=m
+CONFIG_VIDEO_SUN6I_MIPI_CSI2=m
+CONFIG_VIDEO_SUN8I_A83T_MIPI_CSI2=m
+CONFIG_VIDEO_SUN8I_DEINTERLACE=m
+CONFIG_VIDEO_SUN8I_ROTATE=m
+CONFIG_VIDEO_TI_CAL=m
+CONFIG_VIDEO_HANTRO=m
 CONFIG_VIDEO_XILINX=m
+CONFIG_VIDEO_XILINX_CSI2RXSS=m
 CONFIG_VIDEO_XILINX_TPG=m
 CONFIG_SMS_SDIO_DRV=m
 CONFIG_DVB_FIREDTV=m
 CONFIG_PANEL=m
-CONFIG_IMX_IPUV3_CORE=m
 CONFIG_DRM=y
 CONFIG_DRM_LOAD_EDID_FIRMWARE=y
 CONFIG_DRM_DP_AUX_CHARDEV=y
@@ -2814,11 +2835,9 @@ CONFIG_DRM_EXYNOS_DPI=y
 CONFIG_DRM_EXYNOS_DSI=y
 CONFIG_DRM_EXYNOS_HDMI=y
 CONFIG_DRM_EXYNOS_MIC=y
-CONFIG_DRM_EXYNOS_G2D=y
 CONFIG_DRM_EXYNOS_FIMC=y
 CONFIG_DRM_EXYNOS_ROTATOR=y
 CONFIG_DRM_EXYNOS_SCALER=y
-CONFIG_DRM_EXYNOS_GSC=y
 CONFIG_DRM_VMWGFX=m
 CONFIG_DRM_UDL=m
 CONFIG_DRM_AST=m
@@ -2829,12 +2848,10 @@ CONFIG_DRM_RCAR_DW_HDMI=m
 CONFIG_DRM_RZG2L_MIPI_DSI=m
 CONFIG_DRM_SHMOBILE=m
 CONFIG_DRM_SUN4I=m
-CONFIG_DRM_SUN4I_HDMI_CEC=y
 CONFIG_DRM_QXL=m
 CONFIG_DRM_VIRTIO_GPU=y
 CONFIG_DRM_MSM=m
 CONFIG_DRM_TEGRA=m
-CONFIG_DRM_TEGRA_DEBUG=y
 CONFIG_DRM_TEGRA_STAGING=y
 CONFIG_DRM_STM=m
 CONFIG_DRM_STM_DSI=m
@@ -2935,15 +2952,8 @@ CONFIG_DRM_TI_TFP410=m
 CONFIG_DRM_ANALOGIX_ANX78XX=m
 CONFIG_DRM_I2C_ADV7511=m
 CONFIG_DRM_CDNS_DSI=m
-CONFIG_DRM_IMX=m
-CONFIG_DRM_IMX_PARALLEL_DISPLAY=m
-CONFIG_DRM_IMX_TVE=m
-CONFIG_DRM_IMX_LDB=m
-CONFIG_DRM_IMX_HDMI=m
 CONFIG_DRM_IMX_LCDC=m
 CONFIG_DRM_V3D=m
-CONFIG_DRM_VC4=m
-CONFIG_DRM_VC4_HDMI_CEC=y
 CONFIG_DRM_HISI_HIBMC=m
 CONFIG_DRM_HISI_KIRIN=m
 CONFIG_DRM_MEDIATEK=m
@@ -2956,7 +2966,6 @@ CONFIG_DRM_ARCPGU=m
 CONFIG_DRM_BOCHS=m
 CONFIG_DRM_CIRRUS_QEMU=m
 CONFIG_DRM_GM12U320=m
-CONFIG_DRM_OFDRM=m
 CONFIG_DRM_PANEL_MIPI_DBI=m
 CONFIG_DRM_SIMPLEDRM=m
 CONFIG_TINYDRM_HX8357D=m
@@ -2969,12 +2978,9 @@ CONFIG_TINYDRM_REPAPER=m
 CONFIG_TINYDRM_ST7586=m
 CONFIG_TINYDRM_ST7735R=m
 CONFIG_DRM_PL111=m
-CONFIG_DRM_TVE200=m
 CONFIG_DRM_XEN_FRONTEND=m
 CONFIG_DRM_LIMA=m
 CONFIG_DRM_PANFROST=m
-CONFIG_DRM_ASPEED_GFX=m
-CONFIG_DRM_MCDE=m
 CONFIG_DRM_TIDSS=m
 CONFIG_DRM_GUD=m
 CONFIG_DRM_SSD130X=m
@@ -2983,8 +2989,6 @@ CONFIG_DRM_SSD130X_SPI=m
 CONFIG_DRM_SPRD=m
 CONFIG_DRM_PHYTIUM=m
 CONFIG_FB=m
-CONFIG_FB_OMAP2=m
-CONFIG_MMP_DISP=m
 CONFIG_FB_TILEBLITTING=y
 CONFIG_BACKLIGHT_PWM=m
 CONFIG_BACKLIGHT_ADP8860=m
@@ -3065,6 +3069,7 @@ CONFIG_SND_VIRTUOSO=m
 CONFIG_SND_VX222=m
 CONFIG_SND_YMFPCI=m
 CONFIG_SND_HDA_INTEL=y
+CONFIG_SND_HDA_PHYTIUM=m
 CONFIG_SND_HDA_HWDEP=y
 CONFIG_SND_HDA_INPUT_BEEP=y
 CONFIG_SND_HDA_PATCH_LOADER=y
@@ -3348,6 +3353,8 @@ CONFIG_USB_MON=m
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_XHCI_DBGCAP=y
 CONFIG_USB_XHCI_HISTB=m
+CONFIG_USB_XHCI_MTK=m
+CONFIG_USB_XHCI_MVEBU=m
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_OHCI_HCD=y
 CONFIG_USB_UHCI_HCD=m
@@ -3472,10 +3479,21 @@ CONFIG_USB_ISP1301=m
 CONFIG_USB_GADGET=m
 CONFIG_TYPEC=m
 CONFIG_TYPEC_TCPM=m
+CONFIG_TYPEC_TCPCI=m
+CONFIG_TYPEC_RT1711H=m
+CONFIG_TYPEC_TCPCI_MAXIM=m
 CONFIG_TYPEC_FUSB302=m
+CONFIG_TYPEC_QCOM_PMIC=m
 CONFIG_TYPEC_UCSI=m
+CONFIG_UCSI_CCG=m
 CONFIG_UCSI_ACPI=m
+CONFIG_UCSI_STM32G0=m
 CONFIG_TYPEC_TPS6598X=m
+CONFIG_TYPEC_ANX7411=m
+CONFIG_TYPEC_RT1719=m
+CONFIG_TYPEC_HD3SS3220=m
+CONFIG_TYPEC_STUSB160X=m
+CONFIG_TYPEC_WUSB3801=m
 CONFIG_TYPEC_MUX_PI3USB30532=m
 CONFIG_TYPEC_DP_ALTMODE=m
 CONFIG_MMC=m
@@ -3496,13 +3514,21 @@ CONFIG_MMC_DW_EXYNOS=m
 CONFIG_MMC_DW_HI3798CV200=m
 CONFIG_MMC_DW_K3=m
 CONFIG_MMC_DW_PCI=m
+CONFIG_MMC_DW_ROCKCHIP=m
+CONFIG_MMC_SH_MMCIF=m
 CONFIG_MMC_VUB300=m
 CONFIG_MMC_USHC=m
+CONFIG_MMC_USDHI6ROL0=m
 CONFIG_MMC_REALTEK_PCI=m
 CONFIG_MMC_REALTEK_USB=m
+CONFIG_MMC_SUNXI=m
+CONFIG_MMC_HSQ=m
 CONFIG_MMC_TOSHIBA_PCI=m
+CONFIG_MMC_BCM2835=m
 CONFIG_MMC_MTK=m
 CONFIG_MMC_SDHCI_XENON=m
+CONFIG_MMC_SDHCI_AM654=m
+CONFIG_MMC_OWL=m
 CONFIG_MMC_PHYTIUM_SDCI=m
 CONFIG_SCSI_UFSHCD=m
 CONFIG_SCSI_UFSHCD_PCI=m
@@ -3561,8 +3587,24 @@ CONFIG_INFINIBAND_SRPT=m
 CONFIG_INFINIBAND_ISER=m
 CONFIG_INFINIBAND_ISERT=m
 CONFIG_EDAC=y
+CONFIG_EDAC_GHES=m
+CONFIG_EDAC_AL_MC=m
+CONFIG_EDAC_LAYERSCAPE=m
+CONFIG_EDAC_THUNDERX=m
+CONFIG_EDAC_ALTERA=y
+CONFIG_EDAC_ALTERA_SDRAM=y
+CONFIG_EDAC_ALTERA_OCRAM=y
+CONFIG_EDAC_ALTERA_ETHERNET=y
+CONFIG_EDAC_ALTERA_USB=y
+CONFIG_EDAC_ALTERA_QSPI=y
+CONFIG_EDAC_ALTERA_SDMMC=y
+CONFIG_EDAC_SYNOPSYS=m
 CONFIG_EDAC_XGENE=m
 CONFIG_EDAC_PHYTIUM=m
+CONFIG_EDAC_BLUEFIELD=m
+CONFIG_EDAC_DMC520=m
+CONFIG_EDAC_ZYNQMP=m
+CONFIG_EDAC_NPCM=m
 CONFIG_RTC_CLASS=y
 CONFIG_RTC_DRV_ABB5ZES3=m
 CONFIG_RTC_DRV_ABX80X=m
@@ -3592,23 +3634,56 @@ CONFIG_RTC_DRV_ZYNQMP=y
 CONFIG_RTC_DRV_PL030=y
 CONFIG_RTC_DRV_PL031=y
 CONFIG_DMADEVICES=y
-# CONFIG_BCM_SBA_RAID is not set
+CONFIG_ALTERA_MSGDMA=m
+CONFIG_AMBA_PL08X=y
+CONFIG_AXI_DMAC=m
+CONFIG_DMA_BCM2835=m
+CONFIG_DMA_SUN6I=m
+CONFIG_DW_AXI_DMAC=m
+CONFIG_FSL_EDMA=m
+CONFIG_FSL_QDMA=m
+CONFIG_HISI_DMA=m
+CONFIG_IMX_DMA=m
+CONFIG_IMX_SDMA=m
 CONFIG_INTEL_IDMA64=m
 CONFIG_K3_DMA=m
-CONFIG_LPC18XX_DMAMUX=y
-CONFIG_MILBEAUT_HDMAC=m
-CONFIG_MILBEAUT_XDMAC=m
-CONFIG_MMP_PDMA=m
-CONFIG_MMP_TDMA=m
 CONFIG_MV_XOR=y
 CONFIG_MV_XOR_V2=y
 CONFIG_MXS_DMA=y
-CONFIG_NBPFAXI_DMA=m
 CONFIG_OWL_DMA=m
-CONFIG_PCH_DMA=m
 CONFIG_PL330_DMA=m
 CONFIG_PLX_DMA=m
+CONFIG_STM32_DMA=y
+CONFIG_STM32_DMAMUX=y
+CONFIG_STM32_MDMA=y
+CONFIG_SPRD_DMA=m
+CONFIG_TEGRA186_GPC_DMA=m
+CONFIG_TEGRA20_APB_DMA=m
+CONFIG_TEGRA210_ADMA=m
+CONFIG_UNIPHIER_MDMAC=m
+CONFIG_UNIPHIER_XDMAC=m
+CONFIG_XGENE_DMA=m
+CONFIG_XILINX_DMA=m
+CONFIG_XILINX_XDMA=m
+CONFIG_XILINX_ZYNQMP_DMA=m
+CONFIG_XILINX_ZYNQMP_DPDMA=m
 CONFIG_PHYTIUM_DDMA=y
+CONFIG_MTK_HSDMA=m
+CONFIG_MTK_CQDMA=m
+CONFIG_MTK_UART_APDMA=m
+CONFIG_QCOM_BAM_DMA=m
+CONFIG_QCOM_GPI_DMA=m
+CONFIG_QCOM_HIDMA_MGMT=m
+CONFIG_QCOM_HIDMA=m
+CONFIG_DW_DMAC=m
+CONFIG_DW_DMAC_PCI=m
+CONFIG_DW_EDMA=m
+CONFIG_DW_EDMA_PCIE=m
+CONFIG_SF_PDMA=m
+CONFIG_RCAR_DMAC=m
+CONFIG_RENESAS_USB_DMAC=m
+CONFIG_TI_K3_UDMA=m
+CONFIG_TI_K3_UDMA_GLUE_LAYER=m
 CONFIG_ASYNC_TX_DMA=y
 CONFIG_SW_SYNC=y
 CONFIG_UIO_CIF=m
@@ -3633,6 +3708,7 @@ CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES=y
 CONFIG_VHOST_NET=m
 CONFIG_VHOST_SCSI=m
 CONFIG_VHOST_VSOCK=m
+CONFIG_VHOST_CROSS_ENDIAN_LEGACY=y
 CONFIG_HYPERV=m
 CONFIG_HYPERV_UTILS=m
 CONFIG_HYPERV_BALLOON=m
@@ -3660,7 +3736,9 @@ CONFIG_DVB_BUDGET_PATCH=m
 CONFIG_LTE_GDM724X=m
 CONFIG_PI433=m
 CONFIG_QLGE=m
+CONFIG_GOLDFISH=y
 CONFIG_CHROME_PLATFORMS=y
+CONFIG_MELLANOX_PLATFORM=y
 CONFIG_COMMON_CLK_WM831X=m
 CONFIG_COMMON_CLK_MAX77686=m
 CONFIG_COMMON_CLK_SCPI=y
@@ -3671,16 +3749,38 @@ CONFIG_COMMON_CLK_CDCE706=m
 CONFIG_COMMON_CLK_CDCE925=m
 # CONFIG_COMMON_CLK_XGENE is not set
 CONFIG_COMMON_CLK_PWM=m
+CONFIG_HWSPINLOCK=y
+CONFIG_HWSPINLOCK_OMAP=m
+CONFIG_HWSPINLOCK_PHYTIUM=m
+CONFIG_HWSPINLOCK_QCOM=m
+CONFIG_HWSPINLOCK_SPRD=m
+CONFIG_HWSPINLOCK_SUN6I=m
 CONFIG_ARM_MHU=y
+CONFIG_ARM_MHU_V2=m
+CONFIG_IMX_MBOX=m
 CONFIG_PHYTIUM_MBOX=m
 CONFIG_PLATFORM_MHU=y
 CONFIG_PL320_MBOX=y
+CONFIG_ARMADA_37XX_RWTM_MBOX=m
+CONFIG_OMAP2PLUS_MBOX=m
+CONFIG_ROCKCHIP_MBOX=y
+CONFIG_ALTERA_MBOX=m
+CONFIG_BCM2835_MBOX=m
 CONFIG_HI6220_MBOX=m
+CONFIG_QCOM_APCS_IPC=m
+CONFIG_TEGRA_HSP_MBOX=y
+CONFIG_XGENE_SLIMPRO_MBOX=m
+CONFIG_BCM_PDC_MBOX=m
+CONFIG_MTK_ADSP_MBOX=m
+CONFIG_MTK_CMDQ_MBOX=m
+CONFIG_SPRD_MBOX=m
+CONFIG_QCOM_IPCC=m
 CONFIG_IOMMU_IO_PGTABLE_ARMV7S=y
 CONFIG_IOMMU_DEFAULT_PASSTHROUGH=y
 CONFIG_ARM_SMMU=y
 CONFIG_ARM_SMMU_V3=y
 CONFIG_RPMSG_VIRTIO=m
+CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_DEVFREQ_GOV_PERFORMANCE=m
 CONFIG_DEVFREQ_GOV_POWERSAVE=m
 CONFIG_DEVFREQ_GOV_USERSPACE=m
@@ -3714,6 +3814,8 @@ CONFIG_PWM_PCA9685=m
 CONFIG_PWM_PHYTIUM=m
 CONFIG_PWM_TWL=m
 CONFIG_PWM_TWL_LED=m
+CONFIG_AL_FIC=y
+CONFIG_XILINX_INTC=y
 CONFIG_PHYTIUM_IXIC=y
 CONFIG_IPACK_BUS=m
 CONFIG_BOARD_TPCI200=m
@@ -3727,9 +3829,27 @@ CONFIG_PHY_PXA_28NM_USB2=m
 CONFIG_PHY_MAPPHONE_MDM6600=m
 CONFIG_PHY_QCOM_USB_HS=m
 CONFIG_PHY_QCOM_USB_HSIC=m
-CONFIG_PHY_SAMSUNG_USB2=m
 CONFIG_PHY_TUSB1210=m
 CONFIG_POWERCAP=y
+CONFIG_ARM_CCI_PMU=m
+CONFIG_ARM_CCN=m
+CONFIG_ARM_CMN=m
+CONFIG_ARM_SMMU_V3_PMU=m
+CONFIG_ARM_DSU_PMU=m
+CONFIG_FSL_IMX8_DDR_PMU=m
+CONFIG_FSL_IMX9_DDR_PMU=m
+CONFIG_QCOM_L2_PMU=y
+CONFIG_QCOM_L3_PMU=y
+CONFIG_XGENE_PMU=y
+CONFIG_ARM_SPE_PMU=m
+CONFIG_ARM_DMC620_PMU=m
+CONFIG_MARVELL_CN10K_TAD_PMU=m
+CONFIG_APPLE_M1_CPU_PMU=y
+CONFIG_ALIBABA_UNCORE_DRW_PMU=m
+CONFIG_HNS3_PMU=m
+CONFIG_MARVELL_CN10K_DDR_PMU=m
+CONFIG_ARM_CORESIGHT_PMU_ARCH_SYSTEM_PMU=m
+CONFIG_MESON_DDR_PMU=m
 CONFIG_PHYTIUM_PMU=y
 CONFIG_STM=y
 CONFIG_STM_SOURCE_CONSOLE=y
@@ -3933,11 +4053,9 @@ CONFIG_IMA_WRITE_POLICY=y
 # CONFIG_IMA_READ_POLICY is not set
 CONFIG_IMA_APPRAISE=y
 CONFIG_DEFAULT_SECURITY_APPARMOR=y
-CONFIG_INIT_STACK_ALL_ZERO=y
 CONFIG_CRYPTO_USER=m
 # CONFIG_CRYPTO_MANAGER_DISABLE_TESTS is not set
 CONFIG_CRYPTO_PCRYPT=m
-CONFIG_CRYPTO_TEST=m
 CONFIG_CRYPTO_ECDSA=m
 CONFIG_CRYPTO_ECRDSA=m
 CONFIG_CRYPTO_SM2=y
@@ -4011,9 +4129,11 @@ CONFIG_SIGNED_PE_FILE_VERIFICATION=y
 CONFIG_INDIRECT_PIO=y
 CONFIG_CRC4=m
 CONFIG_LIBCRC32C=y
+CONFIG_DMA_CMA=y
 CONFIG_PRINTK_TIME=y
 CONFIG_CONSOLE_LOGLEVEL_DEFAULT=4
 CONFIG_DYNAMIC_DEBUG=y
+# CONFIG_DEBUG_MISC is not set
 CONFIG_DEBUG_INFO_DWARF5=y
 CONFIG_DEBUG_INFO_BTF=y
 CONFIG_GDB_SCRIPTS=y
@@ -4024,6 +4144,7 @@ CONFIG_VMLINUX_MAP=y
 CONFIG_MAGIC_SYSRQ=y
 # CONFIG_SLUB_DEBUG is not set
 # CONFIG_SCHED_DEBUG is not set
+CONFIG_SCHEDSTATS=y
 CONFIG_RCU_CPU_STALL_TIMEOUT=60
 # CONFIG_RCU_TRACE is not set
 # CONFIG_STRICT_DEVMEM is not set
diff --git a/arch/loongarch/configs/deepin_loongarch_desktop_defconfig b/arch/loongarch/configs/deepin_loongarch_desktop_defconfig
index 70755781597d..857af4e997c2 100644
--- a/arch/loongarch/configs/deepin_loongarch_desktop_defconfig
+++ b/arch/loongarch/configs/deepin_loongarch_desktop_defconfig
@@ -1,4 +1,4 @@
-CONFIG_COMPILE_TEST=y
+CONFIG_WERROR=y
 CONFIG_BUILD_SALT="-loong64-desktop-rolling"
 CONFIG_KERNEL_ZSTD=y
 CONFIG_DEFAULT_HOSTNAME=""
@@ -69,7 +69,6 @@ CONFIG_PM_AUTOSLEEP=y
 CONFIG_PM_WAKELOCKS=y
 CONFIG_WQ_POWER_EFFICIENT_DEFAULT=y
 CONFIG_ACPI_SPCR_TABLE=y
-CONFIG_ACPI_EC_DEBUGFS=y
 CONFIG_ACPI_TAD=m
 CONFIG_ACPI_DOCK=y
 CONFIG_ACPI_IPMI=m
@@ -102,6 +101,7 @@ CONFIG_BLK_CGROUP_IOLATENCY=y
 CONFIG_BLK_CGROUP_FC_APPID=y
 CONFIG_BLK_CGROUP_IOCOST=y
 CONFIG_BLK_CGROUP_IOPRIO=y
+# CONFIG_BLK_DEBUG_FS is not set
 CONFIG_BLK_SED_OPAL=y
 CONFIG_BLK_INLINE_ENCRYPTION=y
 CONFIG_BLK_INLINE_ENCRYPTION_FALLBACK=y
@@ -123,7 +123,6 @@ CONFIG_MINIX_SUBPARTITION=y
 CONFIG_SOLARIS_X86_PARTITION=y
 CONFIG_UNIXWARE_DISKLABEL=y
 CONFIG_LDM_PARTITION=y
-CONFIG_LDM_DEBUG=y
 CONFIG_SGI_PARTITION=y
 CONFIG_ULTRIX_PARTITION=y
 CONFIG_SUN_PARTITION=y
@@ -604,7 +603,7 @@ CONFIG_BATMAN_ADV_TRACING=y
 CONFIG_OPENVSWITCH=m
 CONFIG_VSOCKETS=m
 CONFIG_VIRTIO_VSOCKETS=m
-CONFIG_NETLINK_DIAG=m
+CONFIG_NETLINK_DIAG=y
 CONFIG_MPLS_ROUTING=m
 CONFIG_MPLS_IPTUNNEL=m
 CONFIG_HSR=m
@@ -643,7 +642,7 @@ CONFIG_BT_6LOWPAN=m
 CONFIG_BT_LEDS=y
 CONFIG_BT_MSFTEXT=y
 CONFIG_BT_AOSPEXT=y
-CONFIG_BT_SELFTEST=y
+# CONFIG_BT_DEBUGFS is not set
 CONFIG_BT_HCIBTUSB=m
 CONFIG_BT_HCIBTUSB_AUTOSUSPEND=y
 CONFIG_BT_HCIBTUSB_MTK=y
@@ -672,7 +671,6 @@ CONFIG_BT_MRVL_SDIO=m
 CONFIG_BT_ATH3K=m
 CONFIG_BT_MTKSDIO=m
 CONFIG_BT_MTKUART=m
-CONFIG_BT_QCOMSMD=m
 CONFIG_BT_VIRTIO=m
 CONFIG_BT_NXPUART=m
 CONFIG_AF_RXRPC_IPV6=y
@@ -721,6 +719,7 @@ CONFIG_NFC_S3FWRN82_UART=m
 CONFIG_NFC_ST95HF=m
 CONFIG_PAGE_POOL_STATS=y
 CONFIG_PCIEAER=y
+# CONFIG_PCIEASPM is not set
 CONFIG_PCIE_DPC=y
 CONFIG_PCIE_PTM=y
 CONFIG_PCIE_EDR=y
@@ -771,6 +770,7 @@ CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
 # CONFIG_STANDALONE is not set
 # CONFIG_PREVENT_FIRMWARE_BUILD is not set
+# CONFIG_FW_LOADER_DEBUG is not set
 CONFIG_FW_LOADER_COMPRESS=y
 CONFIG_FW_LOADER_COMPRESS_ZSTD=y
 CONFIG_MOXTET=m
@@ -790,7 +790,6 @@ CONFIG_EFI_VARS_PSTORE_DEFAULT_DISABLE=y
 CONFIG_EFI_ZBOOT=y
 CONFIG_EFI_BOOTLOADER_CONTROL=m
 CONFIG_EFI_CAPSULE_LOADER=m
-CONFIG_EFI_TEST=m
 CONFIG_RESET_ATTACK_MITIGATION=y
 CONFIG_EFI_COCO_SECRET=y
 CONFIG_GNSS=m
@@ -884,7 +883,7 @@ CONFIG_ZRAM_MEMORY_TRACKING=y
 CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_DRBD=m
 CONFIG_BLK_DEV_NBD=m
-CONFIG_BLK_DEV_RAM=m
+CONFIG_BLK_DEV_RAM=y
 CONFIG_ATA_OVER_ETH=m
 CONFIG_VIRTIO_BLK=y
 CONFIG_BLK_DEV_RBD=m
@@ -927,7 +926,6 @@ CONFIG_DS1682=m
 CONFIG_LATTICE_ECP3_CONFIG=m
 CONFIG_SRAM=y
 CONFIG_DW_XDATA_PCIE=m
-CONFIG_PCI_ENDPOINT_TEST=m
 CONFIG_XILINX_SDFEC=m
 CONFIG_HISI_HIKEY_USB=m
 CONFIG_OPEN_DICE=m
@@ -956,7 +954,7 @@ CONFIG_GP_PCI1XXXX=m
 CONFIG_BLK_DEV_SD=y
 CONFIG_CHR_DEV_ST=m
 CONFIG_BLK_DEV_SR=m
-CONFIG_CHR_DEV_SG=m
+CONFIG_CHR_DEV_SG=y
 CONFIG_CHR_DEV_SCH=m
 CONFIG_SCSI_ENCLOSURE=m
 CONFIG_SCSI_CONSTANTS=y
@@ -1026,7 +1024,6 @@ CONFIG_SCSI_EFCT=m
 CONFIG_SCSI_DC395x=m
 CONFIG_SCSI_AM53C974=m
 CONFIG_SCSI_WD719X=m
-CONFIG_SCSI_DEBUG=m
 CONFIG_SCSI_PMCRAID=m
 CONFIG_SCSI_PM8001=m
 CONFIG_SCSI_BFA_FC=m
@@ -1133,9 +1130,6 @@ CONFIG_MD_CLUSTER=m
 CONFIG_BCACHE=y
 CONFIG_BCACHE_ASYNC_REGISTRATION=y
 CONFIG_BLK_DEV_DM=y
-CONFIG_DM_DEBUG=y
-CONFIG_DM_DEBUG_BLOCK_MANAGER_LOCKING=y
-CONFIG_DM_DEBUG_BLOCK_STACK_TRACING=y
 CONFIG_DM_UNSTRIPED=y
 CONFIG_DM_CRYPT=y
 CONFIG_DM_SNAPSHOT=y
@@ -1367,6 +1361,7 @@ CONFIG_SKY2=y
 CONFIG_OCTEON_EP=m
 CONFIG_PRESTERA=m
 CONFIG_MLX4_EN=m
+# CONFIG_MLX4_DEBUG is not set
 CONFIG_MLX5_CORE=m
 CONFIG_MLX5_FPGA=y
 CONFIG_MLX5_CORE_EN=y
@@ -1384,6 +1379,7 @@ CONFIG_ENC28J60=m
 CONFIG_ENCX24J600=m
 CONFIG_LAN743X=m
 CONFIG_LAN966X_SWITCH=m
+CONFIG_YT6801=m
 CONFIG_MSCC_OCELOT_SWITCH=m
 CONFIG_MYRI10GE=m
 CONFIG_FEALNX=m
@@ -1416,7 +1412,6 @@ CONFIG_8139TOO=y
 CONFIG_8139TOO_TUNE_TWISTER=y
 CONFIG_8139TOO_8129=y
 CONFIG_R8169=m
-CONFIG_YT6801=m
 CONFIG_ROCKER=m
 CONFIG_SXGBE_ETH=m
 CONFIG_SC92031=m
@@ -1632,13 +1627,13 @@ CONFIG_ATH6KL_SDIO=m
 CONFIG_ATH6KL_USB=m
 CONFIG_AR5523=m
 CONFIG_WIL6210=m
+# CONFIG_WIL6210_DEBUGFS is not set
 CONFIG_ATH10K=m
 CONFIG_ATH10K_PCI=m
 CONFIG_ATH10K_AHB=y
 CONFIG_ATH10K_SDIO=m
 CONFIG_ATH10K_USB=m
 CONFIG_WCN36XX=m
-CONFIG_WCN36XX_DEBUGFS=y
 CONFIG_ATH11K=m
 CONFIG_ATH11K_AHB=m
 CONFIG_ATH11K_PCI=m
@@ -1758,6 +1753,8 @@ CONFIG_RTW88_8723DU=m
 CONFIG_RTW88_8821CE=m
 CONFIG_RTW88_8821CS=m
 CONFIG_RTW88_8821CU=m
+CONFIG_RTW88_DEBUG=y
+CONFIG_RTW88_DEBUGFS=y
 CONFIG_RTW89=m
 CONFIG_RTW89_8851BE=m
 CONFIG_RTW89_8852AE=m
@@ -1766,6 +1763,7 @@ CONFIG_RTW89_8852CE=m
 CONFIG_RTW89_DEBUGMSG=y
 CONFIG_RTW89_DEBUGFS=y
 CONFIG_RSI_91X=m
+# CONFIG_RSI_DEBUGFS is not set
 CONFIG_WFX=m
 CONFIG_CW1200=m
 CONFIG_CW1200_WLAN_SDIO=m
@@ -2215,7 +2213,6 @@ CONFIG_I2C_DESIGNWARE_SLAVE=y
 CONFIG_I2C_DESIGNWARE_PCI=m
 CONFIG_I2C_EMEV2=m
 CONFIG_I2C_GPIO=m
-CONFIG_I2C_GPIO_FAULT_INJECTOR=y
 CONFIG_I2C_KEMPLD=m
 CONFIG_I2C_OCORES=m
 CONFIG_I2C_PCA_PLATFORM=m
@@ -2424,7 +2421,6 @@ CONFIG_MAX8925_POWER=m
 CONFIG_WM831X_BACKUP=m
 CONFIG_WM831X_POWER=m
 CONFIG_WM8350_POWER=m
-CONFIG_TEST_POWER=m
 CONFIG_BATTERY_88PM860X=m
 CONFIG_CHARGER_ADP5061=m
 CONFIG_BATTERY_ACT8945A=m
@@ -2739,6 +2735,7 @@ CONFIG_DA9062_THERMAL=m
 CONFIG_GENERIC_ADC_THERMAL=m
 CONFIG_LOONGSON2_THERMAL=m
 CONFIG_WATCHDOG=y
+CONFIG_WATCHDOG_CORE=y
 CONFIG_WATCHDOG_SYSFS=y
 CONFIG_WATCHDOG_HRTIMER_PRETIMEOUT=y
 CONFIG_WATCHDOG_PRETIMEOUT_GOV=y
@@ -2771,7 +2768,6 @@ CONFIG_STPMIC1_WATCHDOG=m
 CONFIG_ALIM7101_WDT=m
 CONFIG_I6300ESB_WDT=m
 CONFIG_KEMPLD_WDT=m
-CONFIG_LOONGSON1_WDT=m
 CONFIG_MEN_A21_WDT=m
 CONFIG_PCIPCWATCHDOG=m
 CONFIG_WDTPCI=m
@@ -3122,12 +3118,10 @@ CONFIG_MEDIA_DIGITAL_TV_SUPPORT=y
 CONFIG_MEDIA_RADIO_SUPPORT=y
 CONFIG_MEDIA_SDR_SUPPORT=y
 CONFIG_MEDIA_PLATFORM_SUPPORT=y
-CONFIG_VIDEO_ADV_DEBUG=y
 CONFIG_VIDEO_FIXED_MINOR_RANGES=y
 CONFIG_V4L2_FLASH_LED_CLASS=m
 CONFIG_DVB_MMAP=y
 CONFIG_DVB_DEMUX_SECTION_LOSS_LOG=y
-CONFIG_DVB_ULE_DEBUG=y
 CONFIG_MEDIA_USB_SUPPORT=y
 CONFIG_USB_GSPCA=m
 CONFIG_USB_GSPCA_BENQ=m
@@ -3602,7 +3596,6 @@ CONFIG_DRM_ARCPGU=m
 CONFIG_DRM_BOCHS=m
 CONFIG_DRM_CIRRUS_QEMU=m
 CONFIG_DRM_GM12U320=m
-CONFIG_DRM_OFDRM=m
 CONFIG_DRM_PANEL_MIPI_DBI=m
 CONFIG_DRM_SIMPLEDRM=y
 CONFIG_TINYDRM_HX8357D=m
@@ -3717,6 +3710,7 @@ CONFIG_BACKLIGHT_BD6107=m
 CONFIG_BACKLIGHT_ARCXCNN=m
 CONFIG_BACKLIGHT_RAVE_SP=m
 CONFIG_BACKLIGHT_LED=m
+# CONFIG_VGA_CONSOLE is not set
 CONFIG_FRAMEBUFFER_CONSOLE_ROTATION=y
 CONFIG_FRAMEBUFFER_CONSOLE_DEFERRED_TAKEOVER=y
 CONFIG_LOGO=y
@@ -3734,7 +3728,6 @@ CONFIG_SND_SEQ_DUMMY=m
 CONFIG_SND_SEQUENCER_OSS=m
 CONFIG_SND_DUMMY=m
 CONFIG_SND_ALOOP=m
-CONFIG_SND_PCMTEST=m
 CONFIG_SND_VIRMIDI=m
 CONFIG_SND_MTPAV=m
 CONFIG_SND_MTS64=m
@@ -4373,14 +4366,11 @@ CONFIG_USB_SISUSBVGA=m
 CONFIG_USB_LD=m
 CONFIG_USB_TRANCEVIBRATOR=m
 CONFIG_USB_IOWARRIOR=m
-CONFIG_USB_TEST=m
-CONFIG_USB_EHSET_TEST_FIXTURE=m
 CONFIG_USB_ISIGHTFW=m
 CONFIG_USB_YUREX=m
 CONFIG_USB_HUB_USB251XB=m
 CONFIG_USB_HSIC_USB3503=m
 CONFIG_USB_HSIC_USB4604=m
-CONFIG_USB_LINK_LAYER_TEST=m
 CONFIG_USB_CHAOSKEY=m
 CONFIG_USB_ONBOARD_HUB=m
 CONFIG_USB_ATM=m
@@ -4644,6 +4634,7 @@ CONFIG_INFINIBAND_IRDMA=m
 CONFIG_MLX4_INFINIBAND=m
 CONFIG_MLX5_INFINIBAND=m
 CONFIG_INFINIBAND_MTHCA=m
+# CONFIG_INFINIBAND_MTHCA_DEBUG is not set
 CONFIG_INFINIBAND_OCRDMA=m
 CONFIG_INFINIBAND_QEDR=m
 CONFIG_INFINIBAND_VMWARE_PVRDMA=m
@@ -4737,6 +4728,7 @@ CONFIG_RTC_DRV_DS2404=m
 CONFIG_RTC_DRV_DA9052=m
 CONFIG_RTC_DRV_DA9055=m
 CONFIG_RTC_DRV_DA9063=m
+CONFIG_RTC_DRV_EFI=y
 CONFIG_RTC_DRV_STK17TA8=m
 CONFIG_RTC_DRV_M48T86=m
 CONFIG_RTC_DRV_M48T35=m
@@ -4751,7 +4743,7 @@ CONFIG_RTC_DRV_CADENCE=m
 CONFIG_RTC_DRV_FTRTC010=m
 CONFIG_RTC_DRV_PCAP=m
 CONFIG_RTC_DRV_MC13XXX=m
-CONFIG_RTC_DRV_LOONGSON=m
+CONFIG_RTC_DRV_LOONGSON=y
 CONFIG_RTC_DRV_MT6397=m
 CONFIG_RTC_DRV_R7301=m
 CONFIG_RTC_DRV_CPCAP=m
@@ -4821,7 +4813,6 @@ CONFIG_GREYBUS_ES2=m
 CONFIG_COMEDI=m
 CONFIG_COMEDI_MISC_DRIVERS=y
 CONFIG_COMEDI_BOND=m
-CONFIG_COMEDI_TEST=m
 CONFIG_COMEDI_PARPORT=m
 CONFIG_COMEDI_ISA_DRIVERS=y
 CONFIG_COMEDI_PCL711=m
@@ -5062,7 +5053,6 @@ CONFIG_COMMON_CLK_FIXED_MMIO=y
 CONFIG_XILINX_VCU=m
 CONFIG_COMMON_CLK_XLNX_CLKWZRD=m
 CONFIG_HWSPINLOCK=y
-CONFIG_CLKSRC_LOONGSON1_PWM=y
 CONFIG_MAILBOX=y
 CONFIG_PLATFORM_MHU=m
 CONFIG_PCC=y
@@ -5083,6 +5073,7 @@ CONFIG_LOONGSON2_GUTS=m
 CONFIG_LOONGSON2_PM=y
 CONFIG_WPCM450_SOC=m
 CONFIG_QCOM_PMIC_GLINK=m
+CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_DEVFREQ_GOV_PERFORMANCE=m
 CONFIG_DEVFREQ_GOV_POWERSAVE=m
 CONFIG_DEVFREQ_GOV_USERSPACE=m
@@ -5543,10 +5534,8 @@ CONFIG_MCB=m
 CONFIG_MCB_PCI=m
 CONFIG_MCB_LPC=m
 CONFIG_USB4=m
-CONFIG_USB4_DMA_TEST=m
 CONFIG_ANDROID_BINDER_IPC=y
 CONFIG_ANDROID_BINDERFS=y
-CONFIG_ANDROID_BINDER_IPC_SELFTEST=y
 CONFIG_LIBNVDIMM=y
 CONFIG_DEV_DAX=m
 CONFIG_NVMEM_LAYOUT_SL28_VPD=m
@@ -5600,7 +5589,6 @@ CONFIG_FSI_SCOM=m
 CONFIG_FSI_SBEFIFO=m
 CONFIG_FSI_OCC=m
 CONFIG_I2CR_SCOM=m
-CONFIG_TEE=m
 CONFIG_MUX_ADG792A=m
 CONFIG_MUX_ADGS1408=m
 CONFIG_MUX_GPIO=m
@@ -5608,6 +5596,7 @@ CONFIG_MUX_MMIO=m
 CONFIG_SIOX=m
 CONFIG_SIOX_BUS_GPIO=m
 CONFIG_SLIM_QCOM_CTRL=m
+CONFIG_INTERCONNECT=y
 CONFIG_COUNTER=m
 CONFIG_INTERRUPT_CNT=m
 CONFIG_MOST=m
@@ -5617,11 +5606,11 @@ CONFIG_MOST_SND=m
 CONFIG_PECI=m
 CONFIG_HTE=y
 CONFIG_VALIDATE_FS_PARSER=y
-CONFIG_EXT2_FS=m
+CONFIG_EXT2_FS=y
 CONFIG_EXT2_FS_XATTR=y
 CONFIG_EXT2_FS_POSIX_ACL=y
 CONFIG_EXT2_FS_SECURITY=y
-CONFIG_EXT3_FS=m
+CONFIG_EXT3_FS=y
 CONFIG_EXT3_FS_POSIX_ACL=y
 CONFIG_EXT3_FS_SECURITY=y
 CONFIG_REISERFS_FS=m
@@ -5642,7 +5631,7 @@ CONFIG_XFS_ONLINE_REPAIR=y
 CONFIG_GFS2_FS=m
 CONFIG_GFS2_FS_LOCKING_DLM=y
 CONFIG_OCFS2_FS=m
-CONFIG_OCFS2_DEBUG_FS=y
+# CONFIG_OCFS2_DEBUG_MASKLOG is not set
 CONFIG_BTRFS_FS=y
 CONFIG_BTRFS_FS_POSIX_ACL=y
 CONFIG_BTRFS_ASSERT=y
@@ -5661,7 +5650,6 @@ CONFIG_FS_VERITY_BUILTIN_SIGNATURES=y
 CONFIG_FANOTIFY=y
 CONFIG_FANOTIFY_ACCESS_PERMISSIONS=y
 CONFIG_QUOTA_NETLINK_INTERFACE=y
-CONFIG_QUOTA_DEBUG=y
 CONFIG_QFMT_V1=m
 CONFIG_QFMT_V2=y
 CONFIG_AUTOFS_FS=y
@@ -5669,7 +5657,7 @@ CONFIG_FUSE_FS=m
 CONFIG_CUSE=m
 CONFIG_VIRTIO_FS=m
 CONFIG_OVERLAY_FS=m
-CONFIG_FSCACHE=y
+CONFIG_FSCACHE=m
 CONFIG_FSCACHE_STATS=y
 CONFIG_CACHEFILES=m
 CONFIG_CACHEFILES_ONDEMAND=y
@@ -5762,7 +5750,6 @@ CONFIG_NFSD_V4_2_INTER_SSC=y
 CONFIG_NFSD_V4_SECURITY_LABEL=y
 CONFIG_RPCSEC_GSS_KRB5_ENCTYPES_CAMELLIA=y
 CONFIG_RPCSEC_GSS_KRB5_ENCTYPES_AES_SHA2=y
-CONFIG_SUNRPC_DEBUG=y
 CONFIG_CEPH_FS=m
 CONFIG_CEPH_FSCACHE=y
 CONFIG_CEPH_FS_POSIX_ACL=y
@@ -5945,9 +5932,7 @@ CONFIG_CRYPTO_DEV_VIRTIO=m
 CONFIG_CRYPTO_DEV_SAFEXCEL=m
 CONFIG_CRYPTO_DEV_CCREE=m
 CONFIG_CRYPTO_DEV_AMLOGIC_GXL=m
-CONFIG_CRYPTO_DEV_AMLOGIC_GXL_DEBUG=y
 CONFIG_PKCS8_PRIVATE_KEY_PARSER=y
-CONFIG_PKCS7_TEST_KEY=y
 CONFIG_SIGNED_PE_FILE_VERIFICATION=y
 CONFIG_CORDIC=y
 CONFIG_CRYPTO_LIB_CURVE25519=y
@@ -5975,8 +5960,7 @@ CONFIG_FONT_TER16x32=y
 CONFIG_FONT_6x8=y
 CONFIG_PRINTK_TIME=y
 CONFIG_DYNAMIC_DEBUG=y
-CONFIG_DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT=y
-CONFIG_DEBUG_INFO_BTF=y
+# CONFIG_DEBUG_MISC is not set
 CONFIG_DEBUG_INFO_DWARF5=y
 CONFIG_GDB_SCRIPTS=y
 CONFIG_STRIP_ASM_SYMS=y
@@ -5984,20 +5968,15 @@ CONFIG_READABLE_ASM=y
 CONFIG_DEBUG_SECTION_MISMATCH=y
 CONFIG_VMLINUX_MAP=y
 CONFIG_KGDB=y
-CONFIG_DEBUG_PAGE_REF=y
-CONFIG_SHRINKER_DEBUG=y
-CONFIG_SCHED_STACK_END_CHECK=y
-CONFIG_DEBUG_MEMORY_INIT=y
-CONFIG_DEBUG_STACKOVERFLOW=y
+# CONFIG_SLUB_DEBUG is not set
 CONFIG_PANIC_TIMEOUT=30
 CONFIG_HARDLOCKUP_DETECTOR=y
 CONFIG_WQ_WATCHDOG=y
 CONFIG_WQ_CPU_INTENSIVE_REPORT=y
+# CONFIG_SCHED_DEBUG is not set
 CONFIG_SCHEDSTATS=y
-CONFIG_DEBUG_PREEMPT=y
-CONFIG_DEBUG_RT_MUTEXES=y
 CONFIG_RCU_EXP_CPU_STALL_TIMEOUT=20
-CONFIG_RCU_EQS_DEBUG=y
+# CONFIG_RCU_TRACE is not set
 CONFIG_BOOTTIME_TRACING=y
 CONFIG_FUNCTION_GRAPH_RETVAL=y
 CONFIG_FPROBE=y
@@ -6010,7 +5989,6 @@ CONFIG_HWLAT_TRACER=y
 CONFIG_TIMERLAT_TRACER=y
 CONFIG_FTRACE_SYSCALLS=y
 CONFIG_BLK_DEV_IO_TRACE=y
-CONFIG_BPF_KPROBE_OVERRIDE=y
 CONFIG_SYNTH_EVENTS=y
 CONFIG_USER_EVENTS=y
 CONFIG_TRACE_EVENT_INJECT=y
@@ -6018,5 +5996,4 @@ CONFIG_RV=y
 CONFIG_RV_MON_WIP=y
 CONFIG_RV_MON_WWNR=y
 # CONFIG_STRICT_DEVMEM is not set
-CONFIG_FUNCTION_ERROR_INJECTION=y
 # CONFIG_RUNTIME_TESTING_MENU is not set
diff --git a/arch/x86/configs/deepin_x86_desktop_defconfig b/arch/x86/configs/deepin_x86_desktop_defconfig
index c566225c17cd..5d59afaaafab 100644
--- a/arch/x86/configs/deepin_x86_desktop_defconfig
+++ b/arch/x86/configs/deepin_x86_desktop_defconfig
@@ -1,5 +1,4 @@
-CONFIG_COMPILE_TEST=y
-CONFIG_UAPI_HEADER_TEST=y
+CONFIG_WERROR=y
 CONFIG_BUILD_SALT="-amd64-desktop-rolling"
 CONFIG_KERNEL_ZSTD=y
 CONFIG_SYSVIPC=y
@@ -62,7 +61,6 @@ CONFIG_X86_UV=y
 CONFIG_X86_INTEL_MID=y
 CONFIG_X86_INTEL_LPSS=y
 CONFIG_X86_AMD_PLATFORM_DEVICE=y
-CONFIG_IOSF_MBI_DEBUG=y
 CONFIG_PARAVIRT_SPINLOCKS=y
 CONFIG_XEN=y
 CONFIG_XEN_PVH=y
@@ -72,7 +70,6 @@ CONFIG_PROCESSOR_SELECT=y
 CONFIG_GART_IOMMU=y
 CONFIG_X86_REROUTE_FOR_BROKEN_BOOT_IRQS=y
 CONFIG_X86_MCELOG_LEGACY=y
-CONFIG_X86_MCE_INJECT=m
 CONFIG_PERF_EVENTS_INTEL_RAPL=m
 CONFIG_PERF_EVENTS_INTEL_CSTATE=m
 CONFIG_PERF_EVENTS_AMD_POWER=m
@@ -96,15 +93,9 @@ CONFIG_EFI_FAKE_MEMMAP=y
 CONFIG_LIVEPATCH=y
 CONFIG_HIBERNATION=y
 CONFIG_PM_WAKELOCKS=y
-CONFIG_PM_DEBUG=y
-CONFIG_PM_ADVANCED_DEBUG=y
-CONFIG_PM_TRACE_RTC=y
 CONFIG_WQ_POWER_EFFICIENT_DEFAULT=y
 CONFIG_ENERGY_MODEL=y
-CONFIG_ACPI_DEBUGGER=y
-CONFIG_ACPI_DEBUGGER_USER=y
 CONFIG_ACPI_FPDT=y
-CONFIG_ACPI_EC_DEBUGFS=m
 CONFIG_ACPI_TAD=m
 CONFIG_ACPI_DOCK=y
 CONFIG_ACPI_IPMI=m
@@ -119,7 +110,6 @@ CONFIG_ACPI_APEI=y
 CONFIG_ACPI_APEI_GHES=y
 CONFIG_ACPI_APEI_PCIEAER=y
 CONFIG_ACPI_APEI_MEMORY_FAILURE=y
-CONFIG_ACPI_APEI_EINJ=m
 CONFIG_ACPI_DPTF=y
 CONFIG_ACPI_EXTLOG=m
 CONFIG_ACPI_CONFIGFS=m
@@ -168,6 +158,7 @@ CONFIG_BLK_CGROUP_IOLATENCY=y
 CONFIG_BLK_CGROUP_FC_APPID=y
 CONFIG_BLK_CGROUP_IOCOST=y
 CONFIG_BLK_CGROUP_IOPRIO=y
+# CONFIG_BLK_DEBUG_FS is not set
 CONFIG_BLK_SED_OPAL=y
 CONFIG_BLK_INLINE_ENCRYPTION=y
 CONFIG_BLK_INLINE_ENCRYPTION_FALLBACK=y
@@ -182,7 +173,6 @@ CONFIG_MINIX_SUBPARTITION=y
 CONFIG_SOLARIS_X86_PARTITION=y
 CONFIG_UNIXWARE_DISKLABEL=y
 CONFIG_LDM_PARTITION=y
-CONFIG_LDM_DEBUG=y
 CONFIG_SGI_PARTITION=y
 CONFIG_ULTRIX_PARTITION=y
 CONFIG_SUN_PARTITION=y
@@ -205,7 +195,6 @@ CONFIG_MEMORY_HOTREMOVE=y
 CONFIG_KSM=y
 CONFIG_DEFAULT_MMAP_MIN_ADDR=65536
 CONFIG_MEMORY_FAILURE=y
-CONFIG_HWPOISON_INJECT=m
 CONFIG_TRANSPARENT_HUGEPAGE=y
 CONFIG_TRANSPARENT_HUGEPAGE_MADVISE=y
 CONFIG_CMA_AREAS=7
@@ -532,7 +521,6 @@ CONFIG_ATM_LANE=m
 CONFIG_ATM_MPOA=m
 CONFIG_ATM_BR2684=m
 CONFIG_L2TP=m
-CONFIG_L2TP_DEBUGFS=m
 CONFIG_L2TP_V3=y
 CONFIG_L2TP_IP=m
 CONFIG_L2TP_ETH=m
@@ -653,6 +641,7 @@ CONFIG_BT_6LOWPAN=m
 CONFIG_BT_LEDS=y
 CONFIG_BT_MSFTEXT=y
 CONFIG_BT_AOSPEXT=y
+# CONFIG_BT_DEBUGFS is not set
 CONFIG_BT_HCIBTUSB=m
 CONFIG_BT_HCIBTUSB_AUTOSUSPEND=y
 CONFIG_BT_HCIBTUSB_MTK=y
@@ -681,7 +670,6 @@ CONFIG_BT_MRVL_SDIO=m
 CONFIG_BT_ATH3K=m
 CONFIG_BT_MTKSDIO=m
 CONFIG_BT_MTKUART=m
-CONFIG_BT_QCOMSMD=m
 CONFIG_BT_VIRTIO=m
 CONFIG_BT_NXPUART=m
 CONFIG_AF_RXRPC_IPV6=y
@@ -689,7 +677,6 @@ CONFIG_RXKAD=y
 CONFIG_RXPERF=m
 CONFIG_AF_KCM=m
 CONFIG_MCTP=y
-CONFIG_CFG80211_DEBUGFS=y
 CONFIG_MAC80211=m
 CONFIG_MAC80211_MESH=y
 CONFIG_MAC80211_MESSAGE_TRACING=y
@@ -739,7 +726,6 @@ CONFIG_EISA=y
 CONFIG_EISA_VLB_PRIMING=y
 CONFIG_PCI=y
 CONFIG_PCIEAER=y
-CONFIG_PCIEAER_INJECT=m
 CONFIG_PCIE_ECRC=y
 CONFIG_PCIE_DPC=y
 CONFIG_PCIE_PTM=y
@@ -757,8 +743,6 @@ CONFIG_HOTPLUG_PCI_CPCI_ZT5550=m
 CONFIG_HOTPLUG_PCI_CPCI_GENERIC=m
 CONFIG_HOTPLUG_PCI_SHPC=y
 CONFIG_VMD=m
-CONFIG_PCIE_KEEMBAY_HOST=y
-CONFIG_PCIE_KEEMBAY_EP=y
 CONFIG_PCIE_DW_PLAT_HOST=y
 CONFIG_PCIE_DW_PLAT_EP=y
 CONFIG_PCI_ENDPOINT=y
@@ -783,10 +767,10 @@ CONFIG_UEVENT_HELPER=y
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
 # CONFIG_STANDALONE is not set
+# CONFIG_FW_LOADER_DEBUG is not set
 CONFIG_FW_LOADER_COMPRESS=y
 CONFIG_FW_LOADER_COMPRESS_ZSTD=y
 CONFIG_DEBUG_DEVRES=y
-CONFIG_INTEL_IXP4XX_EB=y
 CONFIG_CONNECTOR=y
 CONFIG_EDD=y
 CONFIG_EDD_OFF=y
@@ -797,7 +781,6 @@ CONFIG_GOOGLE_FIRMWARE=y
 CONFIG_EFI_VARS_PSTORE=m
 CONFIG_EFI_BOOTLOADER_CONTROL=m
 CONFIG_EFI_CAPSULE_LOADER=m
-CONFIG_EFI_TEST=m
 CONFIG_RESET_ATTACK_MITIGATION=y
 CONFIG_EFI_RCI2_TABLE=y
 CONFIG_GNSS=m
@@ -1027,7 +1010,6 @@ CONFIG_SCSI_SIM710=m
 CONFIG_SCSI_DC395x=m
 CONFIG_SCSI_AM53C974=m
 CONFIG_SCSI_WD719X=m
-CONFIG_SCSI_DEBUG=m
 CONFIG_SCSI_PMCRAID=m
 CONFIG_SCSI_PM8001=m
 CONFIG_SCSI_BFA_FC=m
@@ -1048,7 +1030,6 @@ CONFIG_SATA_ZPODD=y
 CONFIG_SATA_AHCI=y
 CONFIG_SATA_MOBILE_LPM_POLICY=3
 CONFIG_SATA_AHCI_PLATFORM=m
-CONFIG_SATA_AHCI_SEATTLE=m
 CONFIG_SATA_INIC162X=m
 CONFIG_SATA_ACARD_AHCI=m
 CONFIG_SATA_SIL24=m
@@ -1326,6 +1307,7 @@ CONFIG_SKGE_GENESIS=y
 CONFIG_SKY2=m
 CONFIG_PRESTERA=m
 CONFIG_MLX4_EN=m
+# CONFIG_MLX4_DEBUG is not set
 CONFIG_MLX5_CORE=m
 CONFIG_MLX5_FPGA=y
 CONFIG_MLX5_CORE_EN=y
@@ -1497,15 +1479,11 @@ CONFIG_ADM8211=m
 CONFIG_ATH5K=m
 CONFIG_ATH9K=m
 CONFIG_ATH9K_AHB=y
-CONFIG_ATH9K_DEBUGFS=y
-CONFIG_ATH9K_STATION_STATISTICS=y
 CONFIG_ATH9K_WOW=y
 CONFIG_ATH9K_CHANNEL_CONTEXT=y
 CONFIG_ATH9K_PCI_NO_EEPROM=m
 CONFIG_ATH9K_HTC=m
-CONFIG_ATH9K_HTC_DEBUGFS=y
 CONFIG_ATH9K_HWRNG=y
-CONFIG_ATH9K_COMMON_SPECTRAL=y
 CONFIG_CARL9170=m
 CONFIG_CARL9170_HWRNG=y
 CONFIG_ATH6KL=m
@@ -1514,12 +1492,11 @@ CONFIG_ATH6KL_USB=m
 CONFIG_AR5523=m
 CONFIG_WIL6210=m
 CONFIG_WIL6210_TRACING=y
+# CONFIG_WIL6210_DEBUGFS is not set
 CONFIG_ATH10K=m
 CONFIG_ATH10K_PCI=m
 CONFIG_ATH10K_SDIO=m
 CONFIG_ATH10K_USB=m
-CONFIG_ATH10K_DEBUGFS=y
-CONFIG_ATH10K_SPECTRAL=y
 CONFIG_ATH10K_TRACING=y
 CONFIG_WCN36XX=m
 CONFIG_ATH11K=m
@@ -1549,11 +1526,9 @@ CONFIG_IPW2200_PROMISCUOUS=y
 CONFIG_IPW2200_QOS=y
 CONFIG_IWL4965=m
 CONFIG_IWL3945=m
-CONFIG_IWLEGACY_DEBUGFS=y
 CONFIG_IWLWIFI=m
 CONFIG_IWLDVM=m
 CONFIG_IWLMVM=m
-CONFIG_IWLWIFI_DEBUGFS=y
 # CONFIG_IWLWIFI_DEVICE_TRACING is not set
 CONFIG_HOSTAP=m
 CONFIG_HOSTAP_FIRMWARE=y
@@ -1597,7 +1572,6 @@ CONFIG_MT7615E=m
 CONFIG_MT7663U=m
 CONFIG_MT7663S=m
 CONFIG_MT7915E=m
-CONFIG_MT798X_WMAC=y
 CONFIG_MT7921E=m
 CONFIG_MT7921S=m
 CONFIG_MT7921U=m
@@ -1652,6 +1626,8 @@ CONFIG_RTW89_8851BE=m
 CONFIG_RTW89_8852AE=m
 CONFIG_RTW89_8852BE=m
 CONFIG_RTW89_8852CE=m
+CONFIG_RTW89_DEBUGMSG=y
+CONFIG_RTW89_DEBUGFS=y
 CONFIG_RSI_91X=m
 # CONFIG_RSI_DEBUGFS is not set
 CONFIG_WFX=m
@@ -1684,7 +1660,6 @@ CONFIG_PCI200SYN=m
 CONFIG_WANXL=m
 CONFIG_PC300TOO=m
 CONFIG_FARSYNC=m
-CONFIG_SLIC_DS26522=m
 CONFIG_LAPBETHER=m
 CONFIG_IEEE802154_FAKELB=m
 CONFIG_IEEE802154_AT86RF230=m
@@ -1693,14 +1668,13 @@ CONFIG_IEEE802154_CC2520=m
 CONFIG_IEEE802154_ATUSB=m
 CONFIG_IEEE802154_ADF7242=m
 CONFIG_IEEE802154_CA8210=m
-CONFIG_IEEE802154_CA8210_DEBUGFS=y
 CONFIG_IEEE802154_MCR20A=m
 CONFIG_IEEE802154_HWSIM=m
 CONFIG_WWAN=m
+# CONFIG_WWAN_DEBUGFS is not set
 CONFIG_WWAN_HWSIM=m
 CONFIG_MHI_WWAN_CTRL=m
 CONFIG_MHI_WWAN_MBIM=m
-CONFIG_QCOM_BAM_DMUX=m
 CONFIG_RPMSG_WWAN_CTRL=m
 CONFIG_IOSM=m
 CONFIG_MTK_T7XX=m
@@ -1735,10 +1709,8 @@ CONFIG_KEYBOARD_APPLESPI=m
 CONFIG_KEYBOARD_QT1050=m
 CONFIG_KEYBOARD_QT1070=m
 CONFIG_KEYBOARD_QT2160=m
-CONFIG_KEYBOARD_CLPS711X=m
 CONFIG_KEYBOARD_DLINK_DIR685=m
 CONFIG_KEYBOARD_LKKBD=m
-CONFIG_KEYBOARD_EP93XX=m
 CONFIG_KEYBOARD_GPIO=m
 CONFIG_KEYBOARD_TCA6416=m
 CONFIG_KEYBOARD_TCA8418=m
@@ -1853,7 +1825,6 @@ CONFIG_TOUCHSCREEN_HYCON_HY46XX=m
 CONFIG_TOUCHSCREEN_HYNITRON_CSTXXX=m
 CONFIG_TOUCHSCREEN_ILI210X=m
 CONFIG_TOUCHSCREEN_ILITEK=m
-CONFIG_TOUCHSCREEN_IPROC=m
 CONFIG_TOUCHSCREEN_S6SY761=m
 CONFIG_TOUCHSCREEN_GUNZE=m
 CONFIG_TOUCHSCREEN_EKTF2127=m
@@ -1869,11 +1840,9 @@ CONFIG_TOUCHSCREEN_MSG2638=m
 CONFIG_TOUCHSCREEN_MTOUCH=m
 CONFIG_TOUCHSCREEN_NOVATEK_NVT_TS=m
 CONFIG_TOUCHSCREEN_IMAGIS=m
-CONFIG_TOUCHSCREEN_IMX6UL_TSC=m
 CONFIG_TOUCHSCREEN_INEXIO=m
 CONFIG_TOUCHSCREEN_PENMOUNT=m
 CONFIG_TOUCHSCREEN_EDT_FT5X06=m
-CONFIG_TOUCHSCREEN_MIGOR=m
 CONFIG_TOUCHSCREEN_TOUCHRIGHT=m
 CONFIG_TOUCHSCREEN_TOUCHWIN=m
 CONFIG_TOUCHSCREEN_PIXCIR=m
@@ -1894,7 +1863,6 @@ CONFIG_TOUCHSCREEN_SILEAD=m
 CONFIG_TOUCHSCREEN_SIS_I2C=m
 CONFIG_TOUCHSCREEN_ST1232=m
 CONFIG_TOUCHSCREEN_STMFTS=m
-CONFIG_TOUCHSCREEN_SUN4I=m
 CONFIG_TOUCHSCREEN_SUR40=m
 CONFIG_TOUCHSCREEN_SURFACE3_SPI=m
 CONFIG_TOUCHSCREEN_SX8654=m
@@ -2038,6 +2006,7 @@ CONFIG_HW_RANDOM_INTEL=m
 CONFIG_HW_RANDOM_AMD=m
 CONFIG_HW_RANDOM_BA431=m
 CONFIG_HW_RANDOM_VIA=m
+CONFIG_HW_RANDOM_ZHAOXIN=m
 CONFIG_HW_RANDOM_VIRTIO=m
 CONFIG_HW_RANDOM_XIPHERA=m
 CONFIG_APPLICOM=m
@@ -2047,7 +2016,6 @@ CONFIG_HANGCHECK_TIMER=m
 CONFIG_TCG_TIS_SPI=m
 CONFIG_TCG_TIS_SPI_CR50=y
 CONFIG_TCG_TIS_I2C=m
-CONFIG_TCG_TIS_SYNQUACER=m
 CONFIG_TCG_TIS_I2C_CR50=m
 CONFIG_TCG_TIS_I2C_ATMEL=m
 CONFIG_TCG_TIS_I2C_INFINEON=m
@@ -2134,7 +2102,6 @@ CONFIG_SPI_ZYNQMP_GQSPI=m
 CONFIG_SPI_AMD=m
 CONFIG_SPI_MUX=m
 CONFIG_SPI_SPIDEV=m
-CONFIG_SPI_LOOPBACK_TEST=m
 CONFIG_SPI_TLE62X0=m
 CONFIG_SPI_SLAVE=y
 CONFIG_SPI_SLAVE_TIME=m
@@ -2271,7 +2238,6 @@ CONFIG_MAX8925_POWER=m
 CONFIG_WM831X_BACKUP=m
 CONFIG_WM831X_POWER=m
 CONFIG_WM8350_POWER=m
-CONFIG_TEST_POWER=m
 CONFIG_BATTERY_88PM860X=m
 CONFIG_CHARGER_ADP5061=m
 CONFIG_BATTERY_CW2015=m
@@ -2578,6 +2544,7 @@ CONFIG_INTEL_TCC_COOLING=m
 CONFIG_INTEL_HFI_THERMAL=y
 CONFIG_GENERIC_ADC_THERMAL=m
 CONFIG_WATCHDOG=y
+CONFIG_WATCHDOG_CORE=y
 CONFIG_WATCHDOG_SYSFS=y
 CONFIG_WATCHDOG_PRETIMEOUT_GOV=y
 CONFIG_WATCHDOG_PRETIMEOUT_GOV_PANIC=m
@@ -3089,14 +3056,7 @@ CONFIG_VIDEO_CADENCE_CSI2TX=m
 CONFIG_VIDEO_CAFE_CCIC=m
 CONFIG_VIDEO_VIA_CAMERA=m
 CONFIG_SMS_SDIO_DRV=m
-CONFIG_V4L_TEST_DRIVERS=y
-CONFIG_VIDEO_VIM2M=m
-CONFIG_VIDEO_VICODEC=m
-CONFIG_VIDEO_VIMC=m
-CONFIG_VIDEO_VIVID=m
-CONFIG_VIDEO_VIVID_CEC=y
 CONFIG_DVB_FIREDTV=m
-CONFIG_SMS_SIANO_DEBUGFS=y
 CONFIG_VIDEO_AR0521=m
 CONFIG_VIDEO_HI556=m
 CONFIG_VIDEO_HI846=m
@@ -3227,6 +3187,7 @@ CONFIG_DRM_VMWGFX=m
 CONFIG_DRM_GMA500=m
 CONFIG_DRM_UDL=m
 CONFIG_DRM_AST=m
+CONFIG_DRM_MWV207=m
 CONFIG_DRM_MGAG200=m
 CONFIG_DRM_QXL=m
 CONFIG_DRM_VIRTIO_GPU=m
@@ -3544,7 +3505,6 @@ CONFIG_SND_SOC_INTEL_AVS=m
 CONFIG_SND_SOC_INTEL_AVS_MACH_DA7219=m
 CONFIG_SND_SOC_INTEL_AVS_MACH_DMIC=m
 CONFIG_SND_SOC_INTEL_AVS_MACH_HDAUDIO=m
-CONFIG_SND_SOC_INTEL_AVS_MACH_I2S_TEST=m
 CONFIG_SND_SOC_INTEL_AVS_MACH_MAX98927=m
 CONFIG_SND_SOC_INTEL_AVS_MACH_MAX98357A=m
 CONFIG_SND_SOC_INTEL_AVS_MACH_MAX98373=m
@@ -4033,14 +3993,11 @@ CONFIG_USB_SISUSBVGA=m
 CONFIG_USB_LD=m
 CONFIG_USB_TRANCEVIBRATOR=m
 CONFIG_USB_IOWARRIOR=m
-CONFIG_USB_TEST=m
-CONFIG_USB_EHSET_TEST_FIXTURE=m
 CONFIG_USB_ISIGHTFW=m
 CONFIG_USB_YUREX=m
 CONFIG_USB_HUB_USB251XB=m
 CONFIG_USB_HSIC_USB3503=m
 CONFIG_USB_HSIC_USB4604=m
-CONFIG_USB_LINK_LAYER_TEST=m
 CONFIG_USB_CHAOSKEY=m
 CONFIG_USB_ATM=m
 CONFIG_USB_SPEEDTOUCH=m
@@ -4400,12 +4357,10 @@ CONFIG_VIRTIO_PCI=y
 CONFIG_VIRTIO_VDPA=m
 CONFIG_VIRTIO_PMEM=m
 CONFIG_VIRTIO_BALLOON=y
-CONFIG_VIRTIO_MEM=m
 CONFIG_VIRTIO_INPUT=m
 CONFIG_VIRTIO_MMIO=y
 CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES=y
 CONFIG_VDPA=m
-CONFIG_VDPA_SIM=m
 CONFIG_MLX5_VDPA_NET=m
 CONFIG_VHOST_NET=m
 CONFIG_VHOST_SCSI=m
@@ -4427,7 +4382,6 @@ CONFIG_GREYBUS_ES2=m
 CONFIG_COMEDI=m
 CONFIG_COMEDI_MISC_DRIVERS=y
 CONFIG_COMEDI_BOND=m
-CONFIG_COMEDI_TEST=m
 CONFIG_COMEDI_PARPORT=m
 CONFIG_COMEDI_ISA_DRIVERS=y
 CONFIG_COMEDI_PCL711=m
@@ -4641,8 +4595,8 @@ CONFIG_CROS_EC_ISHTP=m
 CONFIG_CROS_EC_SPI=m
 CONFIG_CROS_EC_LPC=m
 CONFIG_CROS_KBD_LED_BACKLIGHT=m
+# CONFIG_CROS_EC_DEBUGFS is not set
 CONFIG_WILCO_EC=m
-CONFIG_WILCO_EC_DEBUGFS=m
 CONFIG_WILCO_EC_EVENTS=m
 CONFIG_WILCO_EC_TELEMETRY=m
 CONFIG_MELLANOX_PLATFORM=y
@@ -4694,7 +4648,6 @@ CONFIG_IBM_RTL=m
 CONFIG_IDEAPAD_LAPTOP=m
 CONFIG_SENSORS_HDAPS=m
 CONFIG_THINKPAD_ACPI=m
-CONFIG_THINKPAD_ACPI_DEBUGFACILITIES=y
 CONFIG_THINKPAD_LMI=m
 CONFIG_INTEL_ATOMISP2_LED=m
 CONFIG_INTEL_SAR_INT1092=m
@@ -4778,6 +4731,7 @@ CONFIG_RPMSG_VIRTIO=m
 CONFIG_SOUNDWIRE=m
 CONFIG_SOUNDWIRE_QCOM=m
 CONFIG_SOC_TI=y
+CONFIG_DEVFREQ_GOV_SIMPLE_ONDEMAND=y
 CONFIG_DEVFREQ_GOV_PERFORMANCE=y
 CONFIG_DEVFREQ_GOV_POWERSAVE=y
 CONFIG_DEVFREQ_GOV_USERSPACE=y
@@ -5147,29 +5101,15 @@ CONFIG_SERIAL_IPOCTAL=m
 CONFIG_RESET_SIMPLE=y
 CONFIG_RESET_TI_SYSCON=m
 CONFIG_RESET_TI_TPS380X=m
-CONFIG_PHY_PISTACHIO_USB=m
 CONFIG_USB_LGM_PHY=m
-CONFIG_PHY_SUN4I_USB=m
-CONFIG_PHY_SUN9I_USB=m
 CONFIG_BCM_KONA_USB2_PHY=m
-CONFIG_PHY_HI6220_USB=m
-CONFIG_PHY_HI3660_USB=m
-CONFIG_PHY_HI3670_USB=m
-CONFIG_PHY_HISI_INNO_USB2=m
-CONFIG_PHY_INGENIC_USB=m
 CONFIG_PHY_PXA_28NM_HSIC=m
 CONFIG_PHY_PXA_28NM_USB2=m
-CONFIG_PHY_PXA_USB=m
-CONFIG_PHY_MMP3_USB=m
-CONFIG_PHY_MMP3_HSIC=m
 CONFIG_PHY_CPCAP_USB=m
 CONFIG_PHY_QCOM_USB_HS=m
 CONFIG_PHY_QCOM_USB_HSIC=m
-CONFIG_PHY_RALINK_USB=m
 CONFIG_PHY_SAMSUNG_USB2=m
 CONFIG_PHY_TUSB1210=m
-CONFIG_PHY_INTEL_KEEMBAY_EMMC=m
-CONFIG_PHY_INTEL_KEEMBAY_USB=m
 CONFIG_PHY_INTEL_LGM_EMMC=m
 CONFIG_INTEL_RAPL=m
 CONFIG_MCB=m
@@ -5219,6 +5159,7 @@ CONFIG_MUX_GPIO=m
 CONFIG_SIOX=m
 CONFIG_SIOX_BUS_GPIO=m
 CONFIG_SLIM_QCOM_CTRL=m
+CONFIG_INTERCONNECT=y
 CONFIG_104_QUAD_8=m
 CONFIG_MOST=m
 CONFIG_MOST_USB_HDM=m
@@ -5245,6 +5186,7 @@ CONFIG_XFS_ONLINE_REPAIR=y
 CONFIG_GFS2_FS=m
 CONFIG_GFS2_FS_LOCKING_DLM=y
 CONFIG_OCFS2_FS=m
+# CONFIG_OCFS2_DEBUG_MASKLOG is not set
 CONFIG_BTRFS_FS=m
 CONFIG_BTRFS_FS_POSIX_ACL=y
 CONFIG_BTRFS_ASSERT=y
@@ -5263,7 +5205,6 @@ CONFIG_FS_VERITY_BUILTIN_SIGNATURES=y
 CONFIG_FANOTIFY=y
 CONFIG_FANOTIFY_ACCESS_PERMISSIONS=y
 CONFIG_QUOTA_NETLINK_INTERFACE=y
-CONFIG_QUOTA_DEBUG=y
 CONFIG_QFMT_V1=m
 CONFIG_QFMT_V2=m
 CONFIG_AUTOFS_FS=m
@@ -5272,7 +5213,7 @@ CONFIG_CUSE=m
 CONFIG_VIRTIO_FS=m
 CONFIG_OVERLAY_FS=m
 CONFIG_OVERLAY_FS_XINO_AUTO=y
-CONFIG_FSCACHE=y
+CONFIG_FSCACHE=m
 CONFIG_FSCACHE_STATS=y
 CONFIG_CACHEFILES=m
 CONFIG_CACHEFILES_ONDEMAND=y
@@ -5355,7 +5296,6 @@ CONFIG_NFSD_V4_2_INTER_SSC=y
 CONFIG_NFSD_V4_SECURITY_LABEL=y
 # CONFIG_RPCSEC_GSS_KRB5_ENCTYPES_AES_SHA1 is not set
 CONFIG_RPCSEC_GSS_KRB5_ENCTYPES_AES_SHA2=y
-CONFIG_SUNRPC_DEBUG=y
 CONFIG_CEPH_FS=m
 CONFIG_CEPH_FSCACHE=y
 CONFIG_CEPH_FS_POSIX_ACL=y
@@ -5365,6 +5305,7 @@ CONFIG_CIFS=m
 CONFIG_CIFS_UPCALL=y
 CONFIG_CIFS_XATTR=y
 CONFIG_CIFS_POSIX=y
+# CONFIG_CIFS_DEBUG is not set
 CONFIG_CIFS_DFS_UPCALL=y
 CONFIG_CIFS_SWN_UPCALL=y
 CONFIG_CIFS_SMB_DIRECT=y
@@ -5466,11 +5407,9 @@ CONFIG_EVM_EXTRA_SMACK_XATTRS=y
 CONFIG_EVM_ADD_XATTRS=y
 CONFIG_DEFAULT_SECURITY_APPARMOR=y
 CONFIG_LSM="lockdown,yama,integrity,apparmor,selinux,hookmanager"
-CONFIG_INIT_STACK_ALL_ZERO=y
 CONFIG_INIT_ON_ALLOC_DEFAULT_ON=y
 CONFIG_CRYPTO_USER=m
 CONFIG_CRYPTO_PCRYPT=m
-CONFIG_CRYPTO_TEST=m
 CONFIG_CRYPTO_ECRDSA=m
 CONFIG_CRYPTO_SM2=m
 CONFIG_CRYPTO_CURVE25519=m
@@ -5478,6 +5417,7 @@ CONFIG_CRYPTO_AES_TI=m
 CONFIG_CRYPTO_ANUBIS=m
 CONFIG_CRYPTO_BLOWFISH=m
 CONFIG_CRYPTO_CAMELLIA=m
+CONFIG_CRYPTO_DES=m
 CONFIG_CRYPTO_KHAZAD=m
 CONFIG_CRYPTO_SEED=m
 CONFIG_CRYPTO_SM4_GENERIC=m
@@ -5485,8 +5425,10 @@ CONFIG_CRYPTO_TEA=m
 CONFIG_CRYPTO_TWOFISH=m
 CONFIG_CRYPTO_ADIANTUM=m
 CONFIG_CRYPTO_ARC4=m
+CONFIG_CRYPTO_CFB=m
 CONFIG_CRYPTO_KEYWRAP=m
 CONFIG_CRYPTO_LRW=m
+CONFIG_CRYPTO_OFB=m
 CONFIG_CRYPTO_AEGIS128=m
 CONFIG_CRYPTO_GCM=y
 CONFIG_CRYPTO_SEQIV=y
@@ -5550,19 +5492,13 @@ CONFIG_CRYPTO_DEV_CHELSIO=m
 CONFIG_CRYPTO_DEV_VIRTIO=m
 CONFIG_CRYPTO_DEV_SAFEXCEL=m
 CONFIG_CRYPTO_DEV_AMLOGIC_GXL=m
-CONFIG_CRYPTO_DEV_ASPEED=m
-CONFIG_CRYPTO_DEV_ASPEED_HACE_HASH=y
-CONFIG_CRYPTO_DEV_ASPEED_HACE_CRYPTO=y
-CONFIG_CRYPTO_DEV_ASPEED_ACRY=y
 CONFIG_CRYPTO_DEV_TSSE=m
 CONFIG_PKCS8_PRIVATE_KEY_PARSER=m
-CONFIG_PKCS7_TEST_KEY=m
 CONFIG_SIGNED_PE_FILE_VERIFICATION=y
 CONFIG_SYSTEM_EXTRA_CERTIFICATE=y
 CONFIG_SECONDARY_TRUSTED_KEYRING=y
 CONFIG_SYSTEM_BLACKLIST_KEYRING=y
 CONFIG_CRC4=m
-CONFIG_XZ_DEC_TEST=m
 CONFIG_FONTS=y
 CONFIG_FONT_8x8=y
 CONFIG_FONT_6x11=y
@@ -5593,16 +5529,12 @@ CONFIG_KGDB=y
 CONFIG_KGDB_LOW_LEVEL_TRAP=y
 CONFIG_KGDB_KDB=y
 CONFIG_KDB_KEYBOARD=y
-CONFIG_PAGE_POISONING=y
+# CONFIG_SLUB_DEBUG is not set
 CONFIG_DEBUG_WX=y
-CONFIG_DEBUG_STACK_USAGE=y
 CONFIG_SCHED_STACK_END_CHECK=y
-CONFIG_MEMORY_NOTIFIER_ERROR_INJECT=m
 CONFIG_HARDLOCKUP_DETECTOR=y
+# CONFIG_SCHED_DEBUG is not set
 CONFIG_SCHEDSTATS=y
-CONFIG_DEBUG_PLIST=y
-CONFIG_DEBUG_SG=y
-CONFIG_DEBUG_NOTIFIERS=y
 CONFIG_RCU_CPU_STALL_TIMEOUT=60
 # CONFIG_RCU_TRACE is not set
 CONFIG_FUNCTION_PROFILER=y
@@ -5612,22 +5544,12 @@ CONFIG_HWLAT_TRACER=y
 CONFIG_MMIOTRACE=y
 CONFIG_FTRACE_SYSCALLS=y
 CONFIG_BLK_DEV_IO_TRACE=y
-CONFIG_BPF_KPROBE_OVERRIDE=y
 CONFIG_HIST_TRIGGERS=y
-CONFIG_PROVIDE_OHCI1394_DMA_INIT=y
-CONFIG_SAMPLES=y
-CONFIG_SAMPLE_TRACE_PRINTK=m
-CONFIG_SAMPLE_FTRACE_DIRECT=m
-CONFIG_SAMPLE_TRACE_ARRAY=m
-CONFIG_IO_STRICT_DEVMEM=y
+# CONFIG_STRICT_DEVMEM is not set
 # CONFIG_X86_VERBOSE_BOOTUP is not set
 CONFIG_EARLY_PRINTK_DBGP=y
 CONFIG_EARLY_PRINTK_USB_XDBC=y
 CONFIG_IO_DELAY_0XED=y
-CONFIG_DEBUG_BOOT_PARAMS=y
-CONFIG_PUNIT_ATOM_DEBUG=m
+# CONFIG_X86_DEBUG_FPU is not set
 CONFIG_UNWINDER_FRAME_POINTER=y
-CONFIG_NOTIFIER_ERROR_INJECTION=m
-CONFIG_FUNCTION_ERROR_INJECTION=y
-CONFIG_TEST_BPF=m
-CONFIG_TEST_BLACKHOLE_DEV=m
+# CONFIG_RUNTIME_TESTING_MENU is not set
-- 
2.20.1

